---
title: "16SAnalysis"
output: html_document
---

```{r}
library(dada2)
library(dplyr)
library(stringr)
library(phyloseq)
#BiocManager::install("decontam")
library(decontam)
#BiocManager::install("ggtree")
library(ggtree)
library(ape)
#install.packages("picante")
library(picante)
#install.packages("caret")
library(caret)
#install.packages("MLmetrics")
library(MLmetrics)
#install.packages("remotes")
#BiocManager::install("ANCOMBC")
library(ANCOMBC)
#install.packages("ggpubr")
library(ggpubr)
library(rstatix)
#install.packages("plotly")
library(plotly)
```

```{r}
# Directories
fastq.dir = file.path("/home/data/16Sfiles")
trimmed.dir = file.path("/home/data/trimmed")

# Set variables for bash
Sys.setenv(FASTQ_DIR = fastq.dir)
Sys.setenv(TRIMMED_DIR = trimmed.dir)
```

```{bash}
mkdir $TRIMMED_DIR

for FASTQ in $FASTQ_DIR/*_R1_001.fastq.gz
  do
    FASTQ_BASE="$(basename ${FASTQ} '_001.fastq.gz')"
    echo "---------------- TRIMMING: $FASTQ ----------------"
    $HOME/.local/bin/cutadapt -g XTCGCATAGTG -o $TRIMMED_DIR/${FASTQ_BASE}.trim.fastq.gz $FASTQ >>/home/data/cutadaptF.txt
    done

for FASTQ in $FASTQ_DIR/*_R2_001.fastq.gz
  do
    FASTQ_BASE="$(basename ${FASTQ} '_001.fastq.gz')"
    echo "---------------- TRIMMING: $FASTQ ----------------"
    $HOME/.local/bin/cutadapt -g XTCGCATAGGACTAC -o $TRIMMED_DIR/${FASTQ_BASE}.trim.fastq.gz $FASTQ >>/home/data/cutadaptR.txt
    done
```

```{r}
path <- "/home/data/trimmed" 
list.files(path)
```

```{r}
# Forward and reverse fastq filenames have format: SAMPLENAME_R1_001.fastq and SAMPLENAME_R2_001.fastq
fnFs <- sort(list.files(path, pattern="_R1.trim.fastq.gz", full.names = TRUE))
fnRs <- sort(list.files(path, pattern="_R2.trim.fastq.gz", full.names = TRUE))
# Extract sample names, assuming filenames have format: SAMPLENAME_XXX.fastq
sample.names <- str_remove(basename(fnFs), "_L002_R1.trim.fastq.gz")

```

```{r}
# Place filtered files in filtered/ subdirectory
filtFs <- file.path(path, "filtered", paste0(sample.names, "_F_filt.fastq.gz"))
filtRs <- file.path(path, "filtered", paste0(sample.names, "_R_filt.fastq.gz"))
names(filtFs) <- sample.names
names(filtRs) <- sample.names

out <- filterAndTrim(fnFs, filtFs, fnRs, filtRs, truncLen=c(230,190),
              maxN=0, maxEE=c(3,3), trimLeft = c(16,13), truncQ=2, rm.phix=TRUE,
              compress=TRUE, multithread=F) 

errF <- learnErrors(filtFs, nbases = 1e10, multithread=TRUE)
errR <- learnErrors(filtRs, nbases = 1e10, multithread=TRUE)
plotErrors(errF, nominalQ=T)
```

```{r, test code}

dadaFs <- dada(filtFs[1:20], err=errF, multithread=TRUE)
dadaRs <- dada(filtRs[1:20], err=errR, multithread=TRUE)

mergers <- mergePairs(dadaFs, filtFs[1:20], dadaRs, filtRs[1:20], verbose=TRUE)
seqtab <- makeSequenceTable(mergers)

seqtab2 <- seqtab[,nchar(colnames(seqtab)) %in% 248:260]

seqtab.nochim <- removeBimeraDenovo(seqtab2, method="consensus", multithread=TRUE, verbose=TRUE)

table(nchar(getSequences(seqtab.nochim)))

sum(seqtab.nochim)/sum(seqtab)

track2 <- cbind(out[1:20], sapply(dadaFs, getN), sapply(dadaRs, getN), sapply(mergers, getN), rowSums(seqtab.nochim))
# If processing a single sample, remove the sapply calls: e.g. replace sapply(dadaFs, getN) with getN(dadaFs)
colnames(track) <- c("input", "filtered", "denoisedF", "denoisedR", "merged", "nonchim")
rownames(track) <- sample.names
```


```{r}
newFs <- list.files("/home/data/trimmed/filtered", pattern="_F_filt.fastq.gz")

sample.names = str_remove(basename(newFs), "_F_filt.fastq.gz")

mergers <- vector("list", length(sample.names))
names(mergers) <- sample.names
for(sam in sample.names) {
  cat("Processing:", sam, "\n")
    derepF <- derepFastq(filtFs[[sam]])
    ddF <- dada(derepF, err=errF, multithread=TRUE)
    derepR <- derepFastq(filtRs[[sam]])
    ddR <- dada(derepR, err=errR, multithread=TRUE)
    merger <- mergePairs(ddF, derepF, ddR, derepR)
    mergers[[sam]] <- merger
}

rm(derepF); rm(derepR)

seqtab <- makeSequenceTable(mergers)

seqtab2 <- seqtab[,nchar(colnames(seqtab)) %in% 251:255]

rm(seqtab)

saveRDS(seqtab2, "~/seqtab.RDS")

table(nchar(getSequences(seqtab2)))

seqtab.nochim <- removeBimeraDenovo(seqtab2, method="consensus", multithread=TRUE, verbose=TRUE)

saveRDS(seqtab.nochim, "~/seqtab.nochim.RDS")
```

```{r}
getN <- function(x) sum(getUniques(x))
track <- cbind(sapply(dadaFs, getN), sapply(dadaRs, getN), sapply(mergers, getN), rowSums(seqtab.nochim))
# If processing a single sample, remove the sapply calls: e.g. replace sapply(dadaFs, getN) with getN(dadaFs)
colnames(track) <- c("denoisedF", "denoisedR", "merged", "nonchim")
rownames(track) <- sample.names
track =  data.frame(track)
head(track)
```

```{bash, include = F}
wget http://www2.decipher.codes/Classification/TrainingSets/SILVA_SSU_r138_2019.RData -P ./silvadb
```

```{r}
library(DECIPHER)
dna <- DNAStringSet(getSequences(seqtab.nochim)) # Create a DNAStringSet from the ASVs

load("~/silvadb/SILVA_SSU_r138_2019.RData")

ids <- IdTaxa(dna, trainingSet, strand="top", processors=30, verbose=FALSE)

ranks <- c("domain", "phylum", "class", "order", "family", "genus") # ranks of interest

# Convert the output object of class "Taxa" to a matrix analogous to the output from assignTaxonomy
taxid <- t(sapply(ids, function(x) {
        m <- match(ranks, x$rank)
        taxa <- x$taxon[m]
        taxa[startsWith(taxa, "unclassified_")] <- NA
        taxa
}))
colnames(taxid) <- ranks; rownames(taxid) <- getSequences(seqtab.nochim)

saveRDS(taxid, "~/taxid.RDS")
```

```{r}
seqtab.nochim = readRDS("~/seqtab.nochim.RDS")
taxid = readRDS("~/taxid.RDS")
sampleDF = read.csv("~/mouseenv/mouse16SmetaD.csv")

rownames(sampleDF) = sampleDF$SampleName
```

```{r}
colSums(t(seqtab.nochim) != 0)
rowSums(seqtab.nochim)
```

##Creation of tree
```{bash, make a tree with SEPP using gg backbone tree}
wget  "https://raw.github.com/smirarab/sepp-refs/master/gg/sepp-package.tar.bz"

tar xvfj sepp-package.tar.bz

cd sepp-package/sepp

python3 setup.py config -c
```

```{r, make list of sequences in fasta file for tree making}
uniques=getUniques(seqtab.nochim)

sequences=names(uniques)

uniquesToFasta(uniques,fout = "~/mouseenv/seqs.fasta", ids=sequences)
```

```{bash}
./sepp-package/run-sepp.sh sepp-package/test.frag test-gg
```

```{bash}
./sepp-package/run-sepp.sh ./seqs.fasta MOUSE
```

```{r, view tree file with ggtree}
seq.tree = read.tree(file = "~/mouseenv/MOUSE_placement.tog.relabelled.tre") %>%
  keep.tip(., sequences)
```

```{r}
ps <- phyloseq(otu_table(seqtab.nochim, taxa_are_rows=FALSE), 
               sample_data(sampleDF), 
               tax_table(taxid),
              phy_tree(seq.tree))

saveRDS(ps, "~/mouseenv/ps.RDS")

ps = readRDS("~/mouseenv/ps.RDS")
```

```{r}
vegan_otu <- function(physeq) {
    OTU <- otu_table(physeq)
    if (taxa_are_rows(OTU)) {
        OTU <- t(OTU)
    }
    return(as(OTU, "matrix"))
}
```

```{r, remove contaminating sequences}
ps.Bact=subset_taxa(ps, domain=="Bacteria") %>%
  prune_species(speciesSums(.) > 0, .) 

library(ggplot2)

neg = ifelse(sample_data(ps.Bact)$SampleType == "Blank", T, F)

contamdf.prev <- isContaminant(ps.Bact, method="prevalence", neg=neg, threshold = 0.5)

table(contamdf.prev$contaminant)

hist(contamdf.prev$p, 100, ylim = c(0,5000), xlim = c(0,1))

ps.noncontam = prune_taxa(!contamdf.prev$contaminant, ps.Bact)

#### Further remove contaminating sequences
ps.Blank = subset_samples(ps.noncontam, sample_data(ps.noncontam)$SampleType == "Blank") %>%
  prune_species(speciesSums(.) > 0, .) %>%
  filter_taxa(., function(x) sum(x > 3) > (0.25*length(x)), TRUE)

bad_taxa <- ps.Blank %>%
    taxa_sums %>%
    sort(decreasing = TRUE) %>%
    names 

goodTaxa <- setdiff(taxa_names(ps.noncontam), bad_taxa)
ps.noncontam <- prune_taxa(goodTaxa, ps.noncontam)

ps.final = subset_samples(ps.noncontam, sample_data(ps.noncontam)$SampleType != "Blank" & sample_sums(ps.noncontam) > 800) %>%
  prune_species(speciesSums(.) > 0, .) 

min(sample_sums(ps.final2))

# Number ASVs and store the exact sequences in the reference sequence slot - can access with refseq(ps)
dna <- Biostrings::DNAStringSet(taxa_names(ps.final))
names(dna) <- taxa_names(ps.final)
ps.final <- merge_phyloseq(ps.final, dna)
taxa_names(ps.final) <- paste0(seq(ntaxa(ps.final)))
ps.final
remove(dna)
rm(ps.Bact,ps.noncontam,neg,ps, ps.noncontam, contamdf.prev)

saveRDS(ps.final, "~/mouseenv/ps_final.RDS")

ps.final = readRDS("~/mouseenv/ps_final.RDS")
```

```{r, lung NMDS Shannon diversity and richness, updated #############################################}
ps.lung = subset_samples(ps.final, sample_data(ps.final)$SampleType == "Lung" & sample_data(ps.final)$Rep == "A" | sample_data(ps.final)$SampleNumber == "S343") %>%
  subset_taxa(., phylum != "NA") %>%
  prune_species(speciesSums(.) > 0, .) 

min(sample_sums(ps.lung))

ps.lung.r = transform_sample_counts(ps.lung, function(x) x/sum(x)) %>%
  prune_species(speciesSums(.) > 0, .)

y = microbiome::transform(ps.lung, "clr")

z=ordinate(y, method = "RDA", distance = "euclidean") %>%
  plot_ordination(y, ., type="samples", color = "Soil", axes = 2:3) +
  theme_minimal() +
  stat_ellipse() +
  ggtitle("PCA of Lung Samples, Aitchison Distance") +
  facet_grid(~Flu)

PermDF = data.frame(sample_data(y))
BetaD=phyloseq::distance(y, method = "euclidean") 

#permanova by soil
adonis(BetaD~Soil*Flu, permutations = 999, data=PermDF)

plotly::plot_ly(x=z$data$PC1, y=z$data$PC2, z=z$data$PC3, type="scatter3d", mode="markers", color=z$data$Soil)

set.seed(42)
p=ordinate(ps.lung.r, method = "NMDS", distance = "bray", trymax = 1000) %>%
  plot_ordination(ps.lung.r, ., type="samples", color = "Soil") +
  theme_minimal() +
  stat_ellipse() +
  facet_grid(~Flu) 

p$stress #0.22

ordinate(ps.lung.r, method = "PCoA", distance = "bray") %>%
  plot_ordination(ps.lung.r, ., type="samples", color = "Soil") +
  theme_minimal() +
  stat_ellipse() +
  ggtitle("Lung HI microbiome, PCoA, Bray-Curtis")+
  facet_grid(~Flu)

PermDF = data.frame(sample_data(ps.lung.r))
BetaD=vegdist(vegan_otu(ps.lung.r), "bray")

#permanova by soil
adonis(BetaD~Soil*Flu, permutations = 999, data=PermDF)

#Shannon diversity 
Shannon.result = estimate_richness(ps.lung.r, measures = "Shannon")

Shannon.result$Samples = rownames(Shannon.result)
lungDF = sample_data(ps.lung.r)
Shannon = cbind(lungDF, Shannon.result)

hist(Shannon$Shannon)
ggqqplot(Shannon$Shannon)

Shannon$logShannon = log10(max(Shannon$Shannon+1) - Shannon$Shannon)

shapiro.test(Shannon$Shannon)

shapiro.test(Shannon$logShannon)

hist(Shannon$Shannon)
ggqqplot(Shannon$Shannon)
hist(Shannon$logShannon)
ggqqplot(Shannon$logShannon)

fit = aov(logShannon ~ Soil*Flu, data = Shannon)
summary(fit)

TukeyHSD(fit)

r=ggplot(Shannon, aes(Soil, Shannon, fill = Soil)) + 
  geom_boxplot() +
  expand_limits(y=0)  + 
  theme_minimal() +
  xlab(element_blank())+ 
  facet_grid(~Flu) + 
  ylab("Shannon Diversity") 

p = ggplot(Shannon, aes(Soil, logShannon, fill = Soil)) + 
  geom_boxplot() +
  expand_limits(y=0)  + 
  theme_minimal() +
  xlab(element_blank())+ 
  facet_grid(~Flu) + 
  ylab("log(max(SD + 1) - SD") 

#############

Observed.result = estimate_richness(ps.lung, measures = "Observed")

Observed.result$Samples = rownames(Observed.result)

lungDF = sample_data(ps.lung)

Observed = cbind(lungDF, Observed.result)

g=ggplot(Observed, aes(Soil, Observed, fill = Soil)) + 
  geom_boxplot() +
  expand_limits(y=0)  + 
  theme_minimal() +
  xlab(element_blank())+ 
  facet_grid(~Flu) + 
  ylab("Richness")

q = ggarrange(r,g, labels = c("A", "B"), nrow = 2)

hist(Observed$Observed)
ggqqplot(Observed$Observed)

shapiro.test(Observed$Observed)

fit = aov(Observed ~ Soil*Flu, data = Observed)

summary(fit)

TukeyHSD(fit)

m = ggarrange(r, 
              ggarrange(p,g, ncol = 2, labels = c("B", "C")), nrow = 2, labels = "A")
```

```{r, soil day 0 NMDS shannon diversity and richness, updated########################################}
#changed from ps.final
ps.soil = subset_samples(ps.final, sample_data(ps.final)$SampleType == "Soil" & sample_sums(ps.final) > 5000) %>%
  prune_species(speciesSums(.) > 0, .) 

sample_sums(ps.soil)

ps.soil.r = transform_sample_counts(ps.soil, function(x) x/sum(x)) %>%
  prune_species(speciesSums(.) > 0, .)

ps.soil.0 = subset_samples(ps.final, sample_data(ps.final)$SampleType == "Soil" & sample_data(ps.final)$Time == "0") %>%
  prune_species(speciesSums(.) > 0, .) 

#Shannon diversity
Shannon.result = estimate_richness(ps.soil.0, measures = "Shannon")

Shannon.result$Samples = rownames(Shannon.result)
soilDF = sample_data(ps.soil.0)
Shannon = cbind(soilDF, Shannon.result) 

shapiro.test(Shannon$Shannon)
hist(Shannon$Shannon)
Shannon$logShannon = log(Shannon$Shannon)
hist(Shannon$logShannon)

shapiro.test(Shannon$logShannon)

kruskal.test(Shannon ~ Soil, data = Shannon)

ggqqplot(Shannon$Shannon)

fit = aov(Shannon ~ Soil, data = Shannon)
summary(fit)

TukeyHSD(fit)

p=ggplot(Shannon, aes(Soil, Shannon, fill = Soil)) + 
  geom_boxplot() +
  expand_limits(y=0)  + 
  theme_minimal() +
  xlab(element_blank())+
  scale_fill_manual(values = c("#7CAE00", "#00BFC4", "#C77CFF")) +
  ylab("Shannon Diversity")+ 
  theme(legend.position = "none")

#Observed richness 
Observed.result = estimate_richness(ps.soil.0, measures = "Observed")

Observed.result$Samples = rownames(Observed.result)
soilDF = sample_data(ps.soil.0)
Observed = cbind(soilDF, Observed.result) 

g=ggplot(Observed, aes(Soil, Observed, fill = Soil)) + 
  geom_boxplot() +
  expand_limits(y=0)  + 
  theme_minimal() +
  ylab("Richness") +
  scale_fill_manual(values = c("#7CAE00", "#00BFC4", "#C77CFF")) +
  xlab(element_blank())+ 
  theme(legend.position = "none")
Observed$Observed
shapiro.test(Observed$Observed)
hist(Observed$Observed)

kruskal.test(Observed ~ Soil, data = Observed)

fit = aov(Observed ~ Soil, data = Observed)
summary(fit)
plot(fit)

TukeyHSD(fit) #Road-Pine p = 0.02

#######################################

r=ordinate(ps.soil.0, method = "NMDS", distance = "bray", trymax = 1000) %>%
  plot_ordination(ps.soil.0, ., type="samples", color = "Soil") +
  theme_minimal() +
  scale_color_manual(values = c("#7CAE00", "#00BFC4", "#C77CFF")) +
  stat_ellipse()

r$stress #0.09

PermDF = data.frame(sample_data(ps.soil.0))
BetaD=vegdist(vegan_otu(ps.soil.0), "bray")

#permanova by soil
adonis(BetaD~Soil, permutations = 999, data=PermDF)

m = ggarrange(r, 
              ggarrange(p,g, ncol = 2, labels = c("B", "C")), nrow = 2, labels = "A")
```

```{r, day 14 and 35 soils NMDS shannon diversity and richness, updated ################################}
ps.soil.C = subset_samples(ps.final, sample_data(ps.final)$SampleType == "Soil" & sample_data(ps.final)$Time != "0" & sample_sums(ps.final) > 5000) %>%
  prune_species(speciesSums(.) > 0, .)

min(sample_sums(ps.soil.C))

Shannon.result = estimate_richness(ps.soil.C, measures = "Shannon")

Shannon.result$Samples = rownames(Shannon.result)
soilDF = sample_data(ps.soil.C)
Shannon = cbind(soilDF, Shannon.result) 

summary(Shannon$Shannon)

shapiro.test(Shannon$Shannon)

hist(Shannon$Shannon)

Shannon$logShannon = log(Shannon$Shannon)
hist(Shannon$logShannon)

ggqqplot(Shannon$Shannon)

fit = aov(Shannon ~ Soil, data = Shannon)
summary(fit)

TukeyHSD(fit) #Road-Ctrl p = 0.05, River-Ctrl p = 0.06

p=ggplot(Shannon, aes(Soil, Shannon, fill = Soil)) + 
  geom_boxplot() +
  expand_limits(y=0)  + 
  theme_minimal() +
  xlab(element_blank())+
  ylab("Shannon Diversity")+ 
  theme(legend.position = "none")

#Observed richness 
Observed.result = estimate_richness(ps.soil.C, measures = "Observed")

Observed.result$Samples = rownames(Observed.result)
soilDF = sample_data(ps.soil.C)
Observed = cbind(soilDF, Observed.result) 

hist(Observed$Observed)
ggqqplot(Observed$Observed)

Observed$logObserved = log(Observed$Observed)
hist(Observed$logObserved)
ggqqplot(Observed$logObserved)

fit = aov(logObserved ~ Soil, data = Observed)
summary(fit)

TukeyHSD(fit) #Road-Ctrl p = 0.02, Pine-Ctrl p = 0.08

Observed %>%
  pairwise_wilcox_test(Observed ~ Soil, p.adjust.methods = "BH")

g=ggplot(Observed, aes(Soil, logObserved, fill = Soil)) + 
  geom_boxplot() +
  expand_limits(y=0)  + 
  theme_minimal() +
  ylab("Richness") +
  xlab(element_blank())+ 
  theme(legend.position = "none")

#######################################

r=ordinate(ps.soil.C, method = "NMDS", distance = "bray", trymax = 1000) %>%
  plot_ordination(ps.soil, ., type="samples", color = "Soil") +
  theme_minimal() +
  stat_ellipse() 

r$stress #0.15

PermDF = data.frame(sample_data(ps.soil.C))
BetaD=vegdist(vegan_otu(ps.soil.C), "bray")

#permanova by soil
adonis(BetaD~Soil, permutations = 999, data=PermDF)

m = ggarrange(r, 
              ggarrange(p,g, ncol = 2, labels = c("B", "C")), nrow = 2, labels = "A")
```

```{r, shannon correlation with cage-lung, updated #####################}
#######
Shannon.result = estimate_richness(ps.lung, measures = "Shannon")

lungDF = sample_data(ps.lung)
Shannon.lung = cbind(lungDF, Shannon.result)

Shannon.lung$logShannon = log10(max(Shannon.lung$Shannon+1) - Shannon.lung$Shannon)

Shannon.lung$join = paste(Shannon.lung$Soil, Shannon.lung$Cohort, Shannon.lung$Flu)
#######
Shannon.result = estimate_richness(ps.soil, measures = "Shannon")

Shannon.result$Samples = rownames(Shannon.result)
soilDF = sample_data(ps.soil.r)

Shannon.soil = cbind(soilDF, Shannon.result) %>%
  subset(., Time == "35")

Shannon.soil$join = paste(Shannon.soil$Soil, Shannon.soil$Cohort, Shannon.soil$Flu)

Shannon.soil = select(Shannon.soil, Shannon, join)

corDFHI = merge(Shannon.lung, Shannon.soil, by = "join") %>%
  subset(., Flu == "HI")

corDFPR8 = merge(Shannon.lung, Shannon.soil, by = "join") %>%
  subset(., Flu == "PR8")

corDF = rbind(corDFHI, corDFPR8)

g=ggplot(corDF, aes(x = Shannon.x, y = Shannon.y)) + 
  geom_point() + 
  geom_smooth(method = "lm") +
  theme_minimal() + 
  ylab("Cage Shannon diversity")+ 
  xlab("Lung Shannon diversity") +
  facet_grid(~Flu)

r=ggscatter(corDFPR8, x = "Shannon.x", y = "Shannon.y", 
          add = "reg.line", conf.int = TRUE, 
          cor.coef = TRUE, cor.method = "pearson",
          xlab = "Lung PR8 Shannon Diversity", ylab = "Cage Shannon Diversity", cor.coeff.args = list(method = "pearson", label.x.npc = "left", label.y.npc = "bottom"), ggtheme = theme_minimal())

l=ggscatter(corDFHI, x = "Shannon.x", y = "Shannon.y", 
          add = "reg.line", conf.int = TRUE, 
          cor.coef = TRUE, cor.method = "pearson",
          xlab = "Lung HI Shannon Diversity", ylab = "Cage Shannon Diversity", cor.coeff.args = list(method = "pearson", label.x.npc = "left", label.y.npc = "bottom"), ggtheme = theme_minimal())

m=ggarrange(r,l, ncol = 2, labels = c("A", "B"))
```

```{r, observed correlation cage-lung, updated ###########################}
Observed.result = estimate_richness(ps.lung, measures = "Observed")

lungDF = sample_data(ps.lung)
Observed.lung = cbind(lungDF, Observed.result)

Observed.lung$logObserved = log10(max(Observed.lung$Observed+1) - Observed.lung$Observed)

Observed.lung$join = paste(Observed.lung$Soil, Observed.lung$Cohort, Observed.lung$Flu)
Observed.result = estimate_richness(ps.soil, measures = "Observed")

Observed.result$Samples = rownames(Observed.result)
soilDF = sample_data(ps.soil.r)

Observed.soil = cbind(soilDF, Observed.result) %>%
  subset(., Time == "35")

Observed.soil$join = paste(Observed.soil$Soil, Observed.soil$Cohort, Observed.soil$Flu)

Observed.soil = select(Observed.soil, Observed, join)

corDFHI = merge(Observed.lung, Observed.soil, by = "join") %>%
  subset(., Flu == "HI")

corDFPR8 = merge(Observed.lung, Observed.soil, by = "join") %>%
  subset(., Flu == "PR8")

corDF = rbind(corDFHI, corDFPR8)

ggplot(corDFHI, aes(x = Observed.x, y = Observed.y, fill = Soil)) + 
  geom_point() + 
  geom_smooth(method = "lm") +
  theme_minimal() + 
  ylab("Cage Observed diversity")+ 
  xlab("Inverse Lung Observed diversity") +
  facet_grid(~Flu)

p=ggscatter(corDFPR8, x = "Observed.x", y = "Observed.y", 
          add = "reg.line", conf.int = TRUE, 
          cor.coef = TRUE, cor.method = "pearson",
          xlab = "Lung PR8 Richness", ylab = "Cage Richness", cor.coeff.args = list(method = "pearson", label.x.npc = "left", label.y.npc = "bottom"), ggtheme = theme_minimal())

g=ggscatter(corDFHI, x = "Observed.x", y = "Observed.y", 
          add = "reg.line", conf.int = TRUE, 
          cor.coef = TRUE, cor.method = "pearson",
          xlab = "Lung HI Richness", ylab = "Cage Richness", cor.coeff.args = list(method = "pearson", label.x.npc = "left", label.y.npc = "center"), ggtheme = theme_minimal())

m=ggarrange(l,r,g,p, ncol = 2, nrow = 2, labels = c("A", "B", "C", "D"))
```

```{r, fecal NMDS and shannon div and richness, updated ###################################}
ps.fecal = subset_samples(ps.final, sample_data(ps.final)$SampleType == "Fecal") %>%
  subset_taxa(., phylum != "NA") %>%
  prune_species(speciesSums(.) > 0, .) 

ps.fecal = subset_samples(ps.fecal, sample_sums(ps.fecal) > 1000)

min(sample_sums(ps.fecal))

ps.fecal.r = transform_sample_counts(ps.fecal, function(x) x/sum(x)) %>%
  prune_species(speciesSums(.) > 0, .)

fecalDF = sample_data(ps.fecal.r)

sample_data(ps.fecal.r)$Time = as.factor(fecalDF$Time)

set.seed(42)

p=ordinate(ps.fecal.r, method = "NMDS", distance = "bray", trymax = 1000) %>%
  plot_ordination(ps.fecal, ., type="samples", color = "Soil") +
  theme_minimal() +
  stat_ellipse()+ 
  facet_grid(Flu~Time)

p$stress

PermDF = data.frame(sample_data(ps.fecal.r))
BetaD=vegdist(vegan_otu(ps.fecal.r), "bray")

#permanova by soil
adonis(BetaD~Soil+Time+Flu+Soil:Time+Soil:Flu+Time:Flu, permutations = 999, data=PermDF)

PermDF = subset_samples(ps.fecal.r, sample_data(ps.fecal.r)$Time == "35") %>%
  sample_data(.) %>%
  data.frame(.)

BetaD=subset_samples(ps.fecal.r, sample_data(ps.fecal.r)$Time == "35") %>%
  vegan_otu(.) %>%
  vegdist(., "bray")
#permanova by soil
adonis(BetaD~Soil*Flu, permutations = 999, data=PermDF)

##################################
Shannon.result = estimate_richness(ps.fecal.r, measures = "Shannon")

Shannon.result$Samples = rownames(Shannon.result)

fecalDF = sample_data(ps.fecal.r)

Shannon = cbind(fecalDF, Shannon.result)

Shannon$Time = as.factor(Shannon$Time)

hist(Shannon$Shannon)

ggqqplot(Shannon$Shannon)

Shannon$logShannon = log10(max(Shannon$Shannon+1) - Shannon$Shannon)
shapiro.test(Shannon$logShannon)
hist(Shannon$logShannon)

fit = aov(Shannon ~ Soil*Time*Flu, data = Shannon)
summary(fit)

TukeyHSD(fit)

p=ggplot(Shannon, aes(Time, Shannon, fill = Soil)) + 
  geom_boxplot() +
  expand_limits(y=0)  + 
  theme_minimal() +
  xlab(element_blank())+ 
  facet_grid(~Soil) +
  ylab("Shannon Diversity") + 
  xlab("Time (days)")+
  scale_y_continuous(limits = c(0,6)) + 
  theme(legend.position = "none")

####################

Observed.result = estimate_richness(ps.fecal, measures = "Observed")

Observed.result$Samples = rownames(Observed.result)

fecalDF = sample_data(ps.fecal)

Observed = cbind(fecalDF, Observed.result)

Observed$Time = as.factor(Observed$Time)

hist(Observed$Observed)
ggqqplot(Observed$Observed)
shapiro.test(Observed$Observed)

Observed$SQObserved = sqrt(Observed$Observed)
hist(Observed$SQObserved)
ggqqplot(Observed$SQObserved)
shapiro.test(Observed$SQObserved)

g=ggplot(Observed, aes(Time, Observed, fill = Soil)) + 
  geom_boxplot() +
  expand_limits(y=0)  + 
  theme_minimal() +
  xlab(element_blank())+ 
  facet_grid(~Soil) +
  ylab("Richness") + 
  xlab("Time (days)")+
  #scale_y_continuous(limits = c(0,30)) + 
  theme(legend.position = "none")

kruskal.test(Observed ~ Flu, data = Observed)

Observed %>%
  pairwise_wilcox_test(Observed ~ Soil, p.adjust.methods = "bonferroni")

fit = aov(SQObserved ~ Soil*Time, data = Observed)
summary(fit)
TukeyHSD(fit)
plot(fit)

r = ggarrange(p,g, labels = c("A", "B"), nrow = 2)
r
```

```{r, ANCOMBC of the lung, #########################}
sample_data(ps.lung)$Soil = factor(sample_data(ps.lung)$Soil, levels = c("Ctrl", "Pine","River", "Road"))
sample_data(ps.lung)$Flu = as.factor(sample_data(ps.lung)$Flu)
sample_data(ps.lung)$Soil
sample_data(ps.lung)$Flu

out = ancombc(phyloseq = ps.lung, formula = "Soil+Flu", 
              p_adj_method = "BH", zero_cut = 0.95, lib_cut = 1000, 
              group = "Soil", struc_zero = F, neg_lb = F, tol = 1e-5, 
              max_iter = 100, conserve = TRUE, alpha = 0.05, global = TRUE)

out.lung = out
res = out.lung$res
tab_q.lung = res$q

col_name = c("BetaPine_lung", "BetaRiver_lung", "BetaRoad_lung", "BetaFlu_lung")
tab_coef = res$beta
colnames(tab_coef) = col_name
tab_coef$ASV = rownames(tab_coef)

col_name = c("Ctrl-Pine_lung", "Ctrl-River_lung", "Ctrl-Road_lung", "HI-PR8_lung")
colnames(tab_q.lung) = col_name
tab_q.lung
tab_q.lung$ASV = rownames(tab_q.lung)
```

```{r, ANCOMBC of soils day 14 and 35 #############}
sample_data(ps.soil)$Soil = factor(sample_data(ps.soil)$Soil, levels = c("Ctrl", "Pine", "River", "Road"))
sample_data(ps.soil)$Flu = as.factor(sample_data(ps.soil)$Flu)
sample_data(ps.soil)$Soil
sample_data(ps.soil)$Flu

ps.soil.C = subset_samples(ps.soil, sample_data(ps.soil)$Time == "35" | sample_data(ps.soil)$Time == "14") %>%
  prune_species(speciesSums(.) > 0, .)

out = ancombc(phyloseq = ps.soil.C, formula = "Soil + Flu", 
              p_adj_method = "BH", zero_cut = 0.95, lib_cut = 1000, 
              group = "Soil", struc_zero = F, neg_lb = T, tol = 1e-5, 
              max_iter = 100, conserve = TRUE, alpha = 0.05, global = TRUE)

out.soil = out
res = out.soil$res
tab_q.soil = res$q

col_name = c("Ctrl-Pine_soil", "Ctrl-River_soil", "Ctrl-Road_soil", "HI-PR8_soil")
colnames(tab_q.soil) = col_name
tab_q.soil$ASV = rownames(tab_q.soil)
tab_q.soil
```

```{r, ANCOMBC of fecal days 14 and 35 ########################}
sample_data(ps.fecal)$Soil = as.factor(sample_data(ps.fecal)$Soil)
sample_data(ps.fecal)$Flu = as.factor(sample_data(ps.fecal)$Flu)
sample_data(ps.fecal)$Soil
sample_data(ps.fecal)$Flu

ps.fecal.C = subset_samples(ps.fecal, sample_data(ps.fecal)$Time == "35" | sample_data(ps.fecal)$Time == "14") %>%
  prune_species(speciesSums(.) > 0, .)

out = ancombc(phyloseq = ps.fecal.C, formula = "Soil + Flu", 
              p_adj_method = "BH", zero_cut = 0.95, lib_cut = 1000, 
              group = "Soil", struc_zero = F, neg_lb = F, tol = 1e-5, 
              max_iter = 100, conserve = TRUE, alpha = 0.05, global = TRUE)

out.fecal = out
res = out.fecal$res
tab_q.fecal = res$q

col_name = c("Ctrl-Pine_fecal", "Ctrl-River_fecal", "Ctrl-Road_fecal", "HI-PR8_fecal")
colnames(tab_q.fecal) = col_name
tab_q.fecal$ASV = rownames(tab_q.fecal)
tab_q.fecal
```

```{r, combine all q tables and look things over}
tab.fecal = left_join(tab_q.fecal, tab_q.soil, by = "ASV")
tab.lung = left_join(tab_q.lung, tab_q.soil, by = "ASV")

PineDF =  subset(tab.lung, tab.lung$`Ctrl-Pine_lung` < 0.05 & tab.lung$`Ctrl-Pine_soil` < 0.05 ) %>%
  select(., `Ctrl-Pine_lung`,  ASV)
names = PineDF$ASV
PineTax = tax_table(ps.lung.r)[names,] %>%
  data.frame(.)
PineTax$ASV = rownames(PineTax)
PineDF = left_join(PineDF, PineTax, by = "ASV")

RiverDF =  subset(tab.lung, tab.lung$`Ctrl-River_lung` < 0.05 & tab.lung$`Ctrl-River_soil` < 0.05) %>%
  select(., `Ctrl-River_lung`, ASV)
names = RiverDF$ASV
RiverTax = tax_table(ps.lung.r)[names,] %>%
  data.frame(.)
RiverTax$ASV = rownames(RiverTax)
RiverDF = left_join(RiverDF, RiverTax, by = "ASV")

RoadDF =  subset(tab.lung, tab.lung$`Ctrl-Road_lung` < 0.05 & tab.lung$`Ctrl-Road_soil` < 0.05) %>%
  select(., `Ctrl-Road_lung`, ASV)
names = RoadDF$ASV
RoadTax = tax_table(ps.lung.r)[names,] %>%
  data.frame(.)
RoadTax$ASV = rownames(RoadTax)
RoadDF = left_join(RoadDF, RoadTax, by = "ASV")

z = RoadTax$family == "Lachnospiraceae"
sum(z, na.rm = TRUE)

setdiff(PineDF$ASV, bad_taxa_num$ASV)

intersect(PineDF$ASV, RoadDF$ASV)

setdiff(RoadDF$ASV, bad_taxa_num$ASV)

################
PineDF =  subset(tab.fecal, tab.fecal$`Ctrl-Pine_fecal` < 0.05 & tab.fecal$`Ctrl-Pine_soil` < 0.05 ) %>%
  select(., `Ctrl-Pine_fecal`,  ASV)
names = PineDF$ASV
PineTax = tax_table(ps.fecal.C)[names,] %>%
  data.frame(.)
PineTax$ASV = rownames(PineTax)
PineDF = left_join(PineDF, PineTax, by = "ASV")

RiverDF =  subset(tab.fecal, tab.fecal$`Ctrl-River_fecal` < 0.05 & tab.fecal$`Ctrl-River_soil` < 0.05) %>%
  select(., `Ctrl-River_fecal`, ASV)
names = RiverDF$ASV
RiverTax = tax_table(ps.fecal.C)[names,] %>%
  data.frame(.)
RiverTax$ASV = rownames(RiverTax)
RiverDF = left_join(RiverDF, RiverTax, by = "ASV")

RoadDF =  subset(tab.fecal, tab.fecal$`Ctrl-Road_fecal` < 0.05 & tab.fecal$`Ctrl-Road_soil` < 0.05) %>%
  select(., `Ctrl-Road_fecal`, ASV)
names = RoadDF$ASV
RoadTax = tax_table(ps.fecal.C)[names,] %>%
  data.frame(.)
RoadTax$ASV = rownames(RoadTax)
RoadDF = left_join(RoadDF, RoadTax, by = "ASV")

z = RiverTax$family == "Lachnospiraceae"
sum(z, na.rm = TRUE)

setdiff(PineDF$ASV, bad_taxa_num$ASV)

setdiff(RiverDF$ASV, bad_taxa_num$ASV)

intersect(PineDF$ASV, RoadDF$ASV)

setdiff(RoadDF$ASV, bad_taxa_num$ASV)
```

```{r, lach plot ################################}
ps.lung.fam = tax_glom(ps.lung.r, taxrank = "family")

tax_table(ps.lung.fam)[, "family"]

lungdf = sample_data(ps.lung.r)

#lachno

FamDFASV = cbind(lungdf, otu_table(ps.lung.fam)[,"51008"]) %>%
  #subset(., soil == "River") %>%
  select(., Soil, Flu, Time, Ear, Cohort, `51008`) %>%
  #summary(.) %>%
  data.frame(.) 

y_title = base::expression(paste(italic("Lachnospiraceae"), " mean lung relative abundance"))

q=ggplot(FamDFASV, aes(Soil, X51008, fill = Soil)) + 
  geom_boxplot() +
  scale_y_continuous(limits = c(0,1), expand = c(0, 0))+ 
  theme_minimal() +
  xlab(element_blank()) +
  ylab(y_title)+ 
  theme(legend.position = "none") 

FamDFASV %>%
  pairwise_wilcox_test(X51008 ~ Soil, p.adjust.methods = "bonferroni")

#Oscilli

FamDFASV = cbind(lungdf, otu_table(ps.lung.fam)[,"49066"]) %>%
  #subset(., soil == "River") %>%
  select(., Soil, Flu, Time, Ear, Cohort, `49066`) %>%
  #summary(.) %>%
  data.frame(.) 

y_title = base::expression(paste(italic("Oscillospiraceae"), " mean lung relative abundance"))

r=ggplot(FamDFASV, aes(Soil, X49066, fill = Soil)) + 
  geom_boxplot() +
  scale_y_continuous(limits = c(0,0.3), expand = c(0, 0))+ 
  theme_minimal() +
  xlab(element_blank()) +
  ylab(y_title)+ 
  theme(legend.position = "none") 

FamDFASV %>%
  pairwise_wilcox_test(X49066 ~ Soil, p.adjust.methods = "bonferroni")

##############################

ps.fecal.fam =transform_sample_counts(ps.fecal.C, function(x) x/sum(x)) %>%
  tax_glom(., taxrank = "family")

tax_table(ps.fecal.fam)[, "family"]

fecaldf = sample_data(ps.fecal.fam)

#Lachno

FamDFASV = cbind(fecaldf, otu_table(ps.fecal.fam)[,"50665"]) %>%
  #subset(., soil == "River") %>%
  select(., Time, Soil, Flu, Ear, Cohort, `50665`) %>%
  #summary(.) %>%
  data.frame(.) 

y_title = base::expression(paste(italic("Lachnospiraceae"), " mean gut relative abundance"))

l=ggplot(FamDFASV, aes(Soil, X50665, fill = Soil)) + 
  geom_boxplot() +
  scale_y_continuous(limits = c(0,1), expand = c(0, 0))+ 
  theme_minimal() +
  xlab(element_blank()) +
  ylab(y_title)+ 
  theme(legend.position = "none") 

FamDFASV %>%
  pairwise_wilcox_test(X50665 ~ Soil, p.adjust.methods = "bonferroni")

#Oscilli
FamDFASV = cbind(fecaldf, otu_table(ps.fecal.fam)[,"51495"]) %>%
  #subset(., soil == "River") %>%
  select(., Time, Soil, Flu, Ear, Cohort, `51495`) %>%
  #summary(.) %>%
  data.frame(.) 

y_title = base::expression(paste(italic("Oscillospiraceae"), " mean gut relative abundance"))

p=ggplot(FamDFASV, aes(Soil, X51495, fill = Soil)) + 
  geom_boxplot() +
  scale_y_continuous(limits = c(0,0.3), expand = c(0, 0))+ 
  theme_minimal() +
  xlab(element_blank()) +
  ylab(y_title)+ 
  theme(legend.position = "none") 

FamDFASV %>%
  pairwise_wilcox_test(X51495 ~ Soil, p.adjust.methods = "bonferroni")

m=ggarrange(q,l,r,p, ncol = 2, nrow = 2, labels = c("A", "B", "C","D"))
m
```

```{r, find origin of ANCOM ASVs (soil or feces?) Pine ###################################}
PineDF =  subset(tab.fecal, tab.fecal$`Ctrl-Pine_fecal` < 0.05) %>%
  select(., `Ctrl-Pine_fecal`,  ASV)
names = PineDF$ASV
PineTax = tax_table(ps.fecal.C)[names,] %>%
  data.frame(.)
PineTax$ASV = rownames(PineTax)
PineDF = left_join(PineDF, PineTax, by = "ASV")

ps.soil.pine = subset_samples(ps.soil, sample_data(ps.soil)$Soil == "Pine" & sample_data(ps.soil)$Time == "0") %>%
  prune_species(speciesSums(.) > 0, .)

D0Pine <- ps.soil.pine %>%
    taxa_sums %>%
    sort(decreasing = TRUE) %>%
    names

ps.fecalD0 = subset_samples(ps.fecal, sample_data(ps.fecal)$Time == "0") %>%
  prune_species(speciesSums(.) > 0, .)

D0Fecal <- ps.fecalD0 %>%
    taxa_sums %>%
    sort(decreasing = TRUE) %>%
    names

D0FecalPine = setdiff(D0Fecal, D0Pine)

D0FecalPine = setdiff(D0FecalPine, bad_taxa_num$ASV)

D0Pine = setdiff(D0Pine, D0Fecal)

D0Pine = setdiff(D0Pine, bad_taxa_num$ASV)

intersect(PineDF$ASV, D0FecalPine)

# "20749" "47619" "48699" "48734" "48812" "48966" "48970" "49019" "49076" "49084" "49104" "49870" "49894" "49942" "49983" "49999" "50317"
# "50339" "50455" "50463" "50674" "50687" "50691" "50986" "50988" "50998" "50999" "51006" "51028" "51035" "51071" "51095" "51096" "51131"
# "51133" "51332" "51337" "51431" "51435" "51442" "51463" "52018" "52024" "52064" "52068" "52092" "52112" "52154" "52566" "52601" "52603"
# "52613" "52656" "52709" "52756" "52794" "52935"

intersect(PineDF$ASV, D0Pine)

#none

######### Lung #########
PineDF =  subset(tab.lung, tab.lung$`Ctrl-Pine_lung` < 0.05) %>%
  select(., `Ctrl-Pine_lung`,  ASV)
names = PineDF$ASV
PineTax = tax_table(ps.lung.r)[names,] %>%
  data.frame(.)
PineTax$ASV = rownames(PineTax)
PineDF = left_join(PineDF, PineTax, by = "ASV")

ps.soil.pine = subset_samples(ps.soil, sample_data(ps.soil)$Soil == "Pine" & sample_data(ps.soil)$Time == "0") %>%
  prune_species(speciesSums(.) > 0, .)

D0Pine <- ps.soil.pine %>%
    taxa_sums %>%
    sort(decreasing = TRUE) %>%
    names

ps.fecalD0 = subset_samples(ps.fecal, sample_data(ps.fecal)$Time == "0") %>%
  prune_species(speciesSums(.) > 0, .)

D0Fecal <- ps.fecalD0 %>%
    taxa_sums %>%
    sort(decreasing = TRUE) %>%
    names

D0FecalPine = setdiff(D0Fecal, D0Pine)

D0FecalPine = setdiff(D0FecalPine, bad_taxa_num$ASV)

D0Pine = setdiff(D0Pine, D0Fecal)

D0Pine = setdiff(D0Pine, bad_taxa_num$ASV)

intersect(PineDF$ASV, D0FecalPine)

# "49084" "49104" "49983" "49998" "50317" "50384" "50455" "50463" "50601" "50674" "50693" "50798" "50965" "50976" "50986" "50988" "50996"
# "51022" "51028" "51071" "51079" "51121" "51133" "51337" "51421" "51431" "51435" "51503" "52217" "52601" "52603" "52622" "52644" "52656"
# "52705" "52756" "52794" "52836"

intersect(PineDF$ASV, D0Pine)

# "50740" "52839"
```

```{r, find origin of ANCOM ASVs (soil or feces?) Road ###################################}
RoadDF =  subset(tab.fecal, tab.fecal$`Ctrl-Road_fecal` < 0.05) %>%
  select(., `Ctrl-Road_fecal`,  ASV)
names = RoadDF$ASV
RoadTax = tax_table(ps.fecal.C)[names,] %>%
  data.frame(.)
RoadTax$ASV = rownames(RoadTax)
RoadDF = left_join(RoadDF, RoadTax, by = "ASV")

ps.soil.Road = subset_samples(ps.soil, sample_data(ps.soil)$Soil == "Road" & sample_data(ps.soil)$Time == "0") %>%
  prune_species(speciesSums(.) > 0, .)

D0Road <- ps.soil.Road %>%
    taxa_sums %>%
    sort(decreasing = TRUE) %>%
    names

ps.fecalD0 = subset_samples(ps.fecal, sample_data(ps.fecal)$Time == "0") %>%
  prune_species(speciesSums(.) > 0, .)

D0Fecal <- ps.fecalD0 %>%
    taxa_sums %>%
    sort(decreasing = TRUE) %>%
    names

D0FecalRoad = setdiff(D0Fecal, D0Road)

D0FecalRoad = setdiff(D0FecalRoad, bad_taxa_num$ASV)

D0Road = setdiff(D0Road, D0Fecal)

D0Road = setdiff(D0Road, bad_taxa_num$ASV)

intersect(RoadDF$ASV, D0FecalRoad)

# "47539" "49872" "50455" "50463" "51022" "51337" "51377"

intersect(RoadDF$ASV, D0Road)

#none

######### Lung #########
RoadDF =  subset(tab.lung, tab.lung$`Ctrl-Road_lung` < 0.05) %>%
  select(., `Ctrl-Road_lung`,  ASV)
names = RoadDF$ASV
RoadTax = tax_table(ps.lung.r)[names,] %>%
  data.frame(.)
RoadTax$ASV = rownames(RoadTax)
RoadDF = left_join(RoadDF, RoadTax, by = "ASV")

ps.soil.Road = subset_samples(ps.soil, sample_data(ps.soil)$Soil == "Road" & sample_data(ps.soil)$Time == "0") %>%
  prune_species(speciesSums(.) > 0, .)

D0Road <- ps.soil.Road %>%
    taxa_sums %>%
    sort(decreasing = TRUE) %>%
    names

ps.fecalD0 = subset_samples(ps.fecal, sample_data(ps.fecal)$Time == "0") %>%
  prune_species(speciesSums(.) > 0, .)

D0Fecal <- ps.fecalD0 %>%
    taxa_sums %>%
    sort(decreasing = TRUE) %>%
    names

D0FecalRoad = setdiff(D0Fecal, D0Road)

D0FecalRoad = setdiff(D0FecalRoad, bad_taxa_num$ASV)

D0Road = setdiff(D0Road, D0Fecal)

D0Road = setdiff(D0Road, bad_taxa_num$ASV)

intersect(RoadDF$ASV, D0FecalRoad)

# "50384" "50965" "52153" "52504"

intersect(RoadDF$ASV, D0Road)

# "12421" "31878" "44877"
```

```{r, find origin of ANCOM ASVs (soil or feces?) River ##################################}
RiverDF =  subset(tab.fecal, tab.fecal$`Ctrl-River_fecal` < 0.05) %>%
  select(., `Ctrl-River_fecal`, ASV)
names = RiverDF$ASV
RiverTax = tax_table(ps.fecal.C)[names,] %>%
  data.frame(.)
RiverTax$ASV = rownames(RiverTax)
RiverDF = left_join(RiverDF, RiverTax, by = "ASV")

ps.soil.river = subset_samples(ps.soil, sample_data(ps.soil)$Soil == "River" & sample_data(ps.soil)$Time == "0") %>%
  prune_species(speciesSums(.) > 0, .)

D0River <- ps.soil.river %>%
    taxa_sums %>%
    sort(decreasing = TRUE) %>%
    names

ps.fecalD0 = subset_samples(ps.fecal, sample_data(ps.fecal)$Time == "0") %>%
  prune_species(speciesSums(.) > 0, .)

D0Fecal <- ps.fecalD0 %>%
    subset_samples(Time == "0") %>%
    prune_species(speciesSums(.) > 0, .) %>%
    taxa_sums %>%
    sort(decreasing = TRUE) %>%
    names

D0FecalRiver = setdiff(D0Fecal, D0River)

D0FecalRiver = setdiff(D0FecalRiver, bad_taxa_num$ASV)

D0River = setdiff(D0River, D0Fecal)

D0River = setdiff(D0River, bad_taxa_num$ASV)

intersect(RiverDF$ASV, D0FecalRiver)

# "48395" "48695" "48734" "48966" "49084" "49937" "50068" "50384" "50469" "50659" "50674" "50999" "51006" "51022" "51096" 
# "51133" "51336" "51337" "51377" "51431" "51435" "51442" "51446" "51465" "52601" "52603" "52641" "52709" "52933" "52935"

intersect(RiverDF$ASV, D0River)

#none
```

```{r, phylum barplot, #########################################}
library(dplyr)

ps.phylum = tax_glom(ps.lung.r, "phylum")

psC1 = subset_samples(ps.phylum, sample_data(ps.phylum)$Soil == "Pine") 

genusDF = as.data.frame(cbind(t(otu_table(psC1, taxa_are_rows = FALSE)), tax_table(psC1)))

rownames(genusDF) = genusDF$phylum
genusDF$domain = NULL
genusDF$phylum=NULL
genusDF$class=NULL
genusDF$order=NULL
genusDF$family=NULL
genusDF$genus=NULL
genusDF$species = NULL
genusDF = mutate_all(genusDF, function(x) as.numeric(as.character(x)))

genusDF$C1 = rowMeans(genusDF)

PineDF = genusDF$C1

psC1 = subset_samples(ps.phylum, sample_data(ps.phylum)$Soil == "Ctrl") %>%
  tax_glom(., "phylum")

genusDF = as.data.frame(cbind(t(otu_table(psC1, taxa_are_rows = FALSE)), tax_table(psC1)))

rownames(genusDF) = genusDF$phylum
genusDF$domain = NULL
genusDF$phylum=NULL
genusDF$class=NULL
genusDF$order=NULL
genusDF$family=NULL
genusDF$genus=NULL
genusDF$species = NULL
genusDF = mutate_all(genusDF, function(x) as.numeric(as.character(x)))

genusDF$C1 = rowMeans(genusDF)

CtrlDF = genusDF$C1

psC1 = subset_samples(ps.phylum, sample_data(ps.phylum)$Soil == "River") %>%
  tax_glom(., "phylum")

genusDF = as.data.frame(cbind(t(otu_table(psC1, taxa_are_rows = FALSE)), tax_table(psC1)))

rownames(genusDF) = genusDF$phylum
genusDF$domain = NULL
genusDF$phylum=NULL
genusDF$class=NULL
genusDF$order=NULL
genusDF$family=NULL
genusDF$genus=NULL
genusDF$species = NULL
genusDF = mutate_all(genusDF, function(x) as.numeric(as.character(x)))

genusDF$C1 = rowMeans(genusDF)

RiverDF = genusDF$C1

psC1 = subset_samples(ps.phylum, sample_data(ps.phylum)$Soil == "Road") %>%
  tax_glom(., "phylum")

genusDF = as.data.frame(cbind(t(otu_table(psC1, taxa_are_rows = FALSE)), tax_table(psC1)))

rownames(genusDF) = genusDF$phylum
genusDF$domain = NULL
genusDF$phylum=NULL
genusDF$class=NULL
genusDF$order=NULL
genusDF$family=NULL
genusDF$genus=NULL
genusDF$species = NULL
genusDF = mutate_all(genusDF, function(x) as.numeric(as.character(x)))

genusDF$C1 = rowMeans(genusDF)

RoadDF = genusDF$C1

LungAvgDF = cbind(CtrlDF, PineDF, RiverDF, RoadDF) %>%
  data.frame(.)

rownames(LungAvgDF) = rownames(genusDF)

LungAvgDF$Avg = rowMeans(LungAvgDF)

figLung = top_n(LungAvgDF, 5, Avg) 

Other = 1 - colSums(figLung)

figLung = rbind(figLung, Other)

figLung$tax = rownames(figLung)

figLung[6,6] = "Other"

figLung = select(figLung, -Avg) %>%
  dplyr::rename('Ctrl' = CtrlDF,
                'Pine' = PineDF,
                'River' = RiverDF,
                'Road' = RoadDF)

#########Fecal Samples
library(dplyr)

ps.phylum = subset_samples(ps.fecal.r, sample_data(ps.fecal.r)$Time == "14") %>%
  tax_glom(., "phylum")

psC1 = subset_samples(ps.phylum, sample_data(ps.phylum)$Soil == "Pine") 

genusDF = as.data.frame(cbind(t(otu_table(psC1, taxa_are_rows = FALSE)), tax_table(psC1)))

rownames(genusDF) = genusDF$phylum
genusDF$domain = NULL
genusDF$phylum=NULL
genusDF$class=NULL
genusDF$order=NULL
genusDF$family=NULL
genusDF$genus=NULL
genusDF$species = NULL
genusDF = mutate_all(genusDF, function(x) as.numeric(as.character(x)))

genusDF$C1 = rowMeans(genusDF)

PineDF = genusDF$C1

psC1 = subset_samples(ps.phylum, sample_data(ps.phylum)$Soil == "Ctrl") %>%
  tax_glom(., "phylum")

genusDF = as.data.frame(cbind(t(otu_table(psC1, taxa_are_rows = FALSE)), tax_table(psC1)))

rownames(genusDF) = genusDF$phylum
genusDF$domain = NULL
genusDF$phylum=NULL
genusDF$class=NULL
genusDF$order=NULL
genusDF$family=NULL
genusDF$genus=NULL
genusDF$species = NULL
genusDF = mutate_all(genusDF, function(x) as.numeric(as.character(x)))

genusDF$C1 = rowMeans(genusDF)

CtrlDF = genusDF$C1

psC1 = subset_samples(ps.phylum, sample_data(ps.phylum)$Soil == "River") %>%
  tax_glom(., "phylum")

genusDF = as.data.frame(cbind(t(otu_table(psC1, taxa_are_rows = FALSE)), tax_table(psC1)))

rownames(genusDF) = genusDF$phylum
genusDF$domain = NULL
genusDF$phylum=NULL
genusDF$class=NULL
genusDF$order=NULL
genusDF$family=NULL
genusDF$genus=NULL
genusDF$species = NULL
genusDF = mutate_all(genusDF, function(x) as.numeric(as.character(x)))

genusDF$C1 = rowMeans(genusDF)

RiverDF = genusDF$C1

psC1 = subset_samples(ps.phylum, sample_data(ps.phylum)$Soil == "Road") %>%
  tax_glom(., "phylum")

genusDF = as.data.frame(cbind(t(otu_table(psC1, taxa_are_rows = FALSE)), tax_table(psC1)))

rownames(genusDF) = genusDF$phylum
genusDF$domain = NULL
genusDF$phylum=NULL
genusDF$class=NULL
genusDF$order=NULL
genusDF$family=NULL
genusDF$genus=NULL
genusDF$species = NULL
genusDF = mutate_all(genusDF, function(x) as.numeric(as.character(x)))

genusDF$C1 = rowMeans(genusDF)

RoadDF = genusDF$C1

FecalAvgDF = cbind(CtrlDF, PineDF, RiverDF, RoadDF) %>%
  data.frame(.)

rownames(FecalAvgDF) = rownames(genusDF)

FecalAvgDF$Avg = rowMeans(FecalAvgDF)

figFecal = top_n(FecalAvgDF, 5, Avg) 

Other = 1 - colSums(figFecal)

figFecal = rbind(figFecal, Other)

figFecal$tax = rownames(figFecal)

figFecal[6,6] = "Other"

figFecal = select(figFecal, -Avg) %>%
  dplyr::rename('Ctrl' = CtrlDF,
                'Pine' = PineDF,
                'River' = RiverDF,
                'Road' = RoadDF)

#########Soil Samples
library(dplyr)

ps.phylum = subset_samples(ps.soil.r, sample_data(ps.soil.r)$Time == "14") %>%
  tax_glom(., "phylum")

psC1 = subset_samples(ps.phylum, sample_data(ps.phylum)$Soil == "Pine") 

genusDF = as.data.frame(cbind(t(otu_table(psC1, taxa_are_rows = FALSE)), tax_table(psC1)))

rownames(genusDF) = genusDF$phylum
genusDF$domain = NULL
genusDF$phylum=NULL
genusDF$class=NULL
genusDF$order=NULL
genusDF$family=NULL
genusDF$genus=NULL
genusDF$species = NULL
genusDF = mutate_all(genusDF, function(x) as.numeric(as.character(x)))

genusDF$C1 = rowMeans(genusDF)

PineDF = genusDF$C1

psC1 = subset_samples(ps.phylum, sample_data(ps.phylum)$Soil == "Ctrl") %>%
  tax_glom(., "phylum")

genusDF = as.data.frame(cbind(t(otu_table(psC1, taxa_are_rows = FALSE)), tax_table(psC1)))

rownames(genusDF) = genusDF$phylum
genusDF$domain = NULL
genusDF$phylum=NULL
genusDF$class=NULL
genusDF$order=NULL
genusDF$family=NULL
genusDF$genus=NULL
genusDF$species = NULL
genusDF = mutate_all(genusDF, function(x) as.numeric(as.character(x)))

genusDF$C1 = rowMeans(genusDF)

CtrlDF = genusDF$C1

psC1 = subset_samples(ps.phylum, sample_data(ps.phylum)$Soil == "River") %>%
  tax_glom(., "phylum")

genusDF = as.data.frame(cbind(t(otu_table(psC1, taxa_are_rows = FALSE)), tax_table(psC1)))

rownames(genusDF) = genusDF$phylum
genusDF$domain = NULL
genusDF$phylum=NULL
genusDF$class=NULL
genusDF$order=NULL
genusDF$family=NULL
genusDF$genus=NULL
genusDF$species = NULL
genusDF = mutate_all(genusDF, function(x) as.numeric(as.character(x)))

genusDF$C1 = rowMeans(genusDF)

RiverDF = genusDF$C1

psC1 = subset_samples(ps.phylum, sample_data(ps.phylum)$Soil == "Road") %>%
  tax_glom(., "phylum")

genusDF = as.data.frame(cbind(t(otu_table(psC1, taxa_are_rows = FALSE)), tax_table(psC1)))

rownames(genusDF) = genusDF$phylum
genusDF$domain = NULL
genusDF$phylum=NULL
genusDF$class=NULL
genusDF$order=NULL
genusDF$family=NULL
genusDF$genus=NULL
genusDF$species = NULL
genusDF = mutate_all(genusDF, function(x) as.numeric(as.character(x)))

genusDF$C1 = rowMeans(genusDF)

RoadDF = genusDF$C1

SoilAvgDF = cbind(CtrlDF, PineDF, RiverDF, RoadDF) %>%
  data.frame(.)

rownames(SoilAvgDF) = rownames(genusDF)

SoilAvgDF$Avg = rowMeans(SoilAvgDF)

figSoil = top_n(SoilAvgDF, 5, Avg) 

Other = 1 - colSums(figSoil)

figSoil = rbind(figSoil, Other)

figSoil$tax = rownames(figSoil)

figSoil[6,6] = "Other"

figSoil = select(figSoil, -Avg) %>%
  dplyr::rename('Ctrl' = CtrlDF,
                'Pine' = PineDF,
                'River' = RiverDF,
                'Road' = RoadDF)

#############Put it together
library(reshape2)
fLung = melt(figLung)
fLung$Sample = "Lung"
fSoil = melt(figSoil)
fSoil$Sample = "Soil"
fFecal = melt(figFecal)
fFecal$Sample = "Fecal"

f = rbind(fLung, fSoil, fFecal)
unique(f$tax)
library(RColorBrewer)

mycolors = c("#666666","#4A72A6","#B75F49","#5ab4ac","#e6550d","#FFB716","#E41A1C","#48A462")#,"#e6550d","#B75F49","#5ab4ac","#666666")

f$tax = factor(f$tax, levels=c("Other","Proteobacteria", "Planctomycetota", "Verrucomicrobiota", "Acidobacteriota", "Actinobacteriota", "Bacteroidota", "Firmicutes"))

p=ggplot(f, aes(fill = tax, y=value, x=variable)) +
  geom_bar(position="stack", stat="identity", color = "black", size = 0.25) +
  scale_fill_manual(values = mycolors) +
  xlab(NULL) +
  ylab("Mean relative abundance") +
  labs(fill = "Phylum") +
  theme_minimal() +
  scale_y_continuous(limits = c(0,1), expand = c(0, 0))+
theme(legend.text.align = 0) +
  facet_grid(~Sample)
p

```


####################################################

```{r, ANCOM-BC for River lung flu samples########}
sample_data(ps.lung)$Soil = factor(sample_data(ps.lung)$Soil, levels = c("Ctrl", "Pine","River", "Road"))
sample_data(ps.lung)$Flu = as.factor(sample_data(ps.lung)$Flu)
sample_data(ps.lung)$Soil
sample_data(ps.lung)$Flu

ps.lungRiver = subset_samples(ps.lung, sample_data(ps.lung)$Soil == "River") %>%
    prune_species(speciesSums(.) > 0, .)

out = ancombc(phyloseq = ps.lungRiver, formula = "Flu", 
              p_adj_method = "BH", zero_cut = 0.95, lib_cut = 1000, 
              group = "Flu", struc_zero = F, neg_lb = F, tol = 1e-5, 
              max_iter = 100, conserve = TRUE, alpha = 0.05, global = TRUE)

out.lungRiver = out
res = out.lungRiver$res
tab_q.lungRiver = res$q 

col_name = c("River_Flu_lung")
colnames(tab_q.lungRiver) = col_name
tab_q.lungRiver
tab_q.lungRiver$ASV = rownames(tab_q.lungRiver)

FluDF = subset(tab_q.lungRiver, tab_q.lungRiver$River_Flu_lung < 0.05)

names = FluDF$ASV
FluTax = tax_table(ps.lung)[names,] %>%
  data.frame(.)
FluTax$ASV = rownames(FluTax)
FluDF = left_join(FluDF, FluTax, by = "ASV")
RiverDFFlu = FluDF
```

```{r, ANCOM-BC for Pine lung flu samples#########}
sample_data(ps.lung)$Soil = factor(sample_data(ps.lung)$Soil, levels = c("Ctrl", "Pine","River", "Road"))
sample_data(ps.lung)$Flu = as.factor(sample_data(ps.lung)$Flu)
sample_data(ps.lung)$Soil
sample_data(ps.lung)$Flu

ps.lungRiver = subset_samples(ps.lung, sample_data(ps.lung)$Soil == "Pine") %>%
    prune_species(speciesSums(.) > 0, .)

out = ancombc(phyloseq = ps.lungRiver, formula = "Flu", 
              p_adj_method = "BH", zero_cut = 0.95, lib_cut = 1000, 
              group = "Flu", struc_zero = F, neg_lb = F, tol = 1e-5, 
              max_iter = 100, conserve = TRUE, alpha = 0.05, global = TRUE)

out.lungRiver = out
res = out.lungRiver$res
tab_q.lungRiver = res$q 

col_name = c("River_Flu_lung")
colnames(tab_q.lungRiver) = col_name
tab_q.lungRiver
tab_q.lungRiver$ASV = rownames(tab_q.lungRiver)

FluDF = subset(tab_q.lungRiver, tab_q.lungRiver$River_Flu_lung < 0.05)

names = FluDF$ASV
FluTax = tax_table(ps.lung)[names,] %>%
  data.frame(.)
FluTax$ASV = rownames(FluTax)
FluDF = left_join(FluDF, FluTax, by = "ASV")
PineDFFlu = FluDF
```

```{r, ANCOM-BC for Road lung flu samples#########}
sample_data(ps.lung)$Soil = factor(sample_data(ps.lung)$Soil, levels = c("Ctrl", "Pine","River", "Road"))
sample_data(ps.lung)$Flu = as.factor(sample_data(ps.lung)$Flu)
sample_data(ps.lung)$Soil
sample_data(ps.lung)$Flu

ps.lungRiver = subset_samples(ps.lung, sample_data(ps.lung)$Soil == "Road") %>%
    prune_species(speciesSums(.) > 0, .)

out = ancombc(phyloseq = ps.lungRiver, formula = "Flu", 
              p_adj_method = "BH", zero_cut = 0.95, lib_cut = 1000, 
              group = "Flu", struc_zero = F, neg_lb = F, tol = 1e-5, 
              max_iter = 100, conserve = TRUE, alpha = 0.05, global = TRUE)

out.lungRiver = out
res = out.lungRiver$res
tab_q.lungRiver = res$q 

col_name = c("River_Flu_lung")
colnames(tab_q.lungRiver) = col_name
tab_q.lungRiver
tab_q.lungRiver$ASV = rownames(tab_q.lungRiver)

FluDF = subset(tab_q.lungRiver, tab_q.lungRiver$River_Flu_lung < 0.05)

names = FluDF$ASV
FluTax = tax_table(ps.lung)[names,] %>%
  data.frame(.)
FluTax$ASV = rownames(FluTax)
FluDF = left_join(FluDF, FluTax, by = "ASV")
RoadDFFlu = FluDF
```

```{r, ANCOM-BC for Ctrl lung flu samples#########}
sample_data(ps.lung)$Soil = factor(sample_data(ps.lung)$Soil, levels = c("Ctrl", "Pine","River", "Road"))
sample_data(ps.lung)$Flu = as.factor(sample_data(ps.lung)$Flu)
sample_data(ps.lung)$Soil
sample_data(ps.lung)$Flu

ps.lungRiver = subset_samples(ps.lung, sample_data(ps.lung)$Soil == "Ctrl") %>%
    prune_species(speciesSums(.) > 0, .)

out = ancombc(phyloseq = ps.lungRiver, formula = "Flu", 
              p_adj_method = "BH", zero_cut = 0.95, lib_cut = 1000, 
              group = "Flu", struc_zero = F, neg_lb = F, tol = 1e-5, 
              max_iter = 100, conserve = TRUE, alpha = 0.05, global = TRUE)

out.lungRiver = out
res = out.lungRiver$res
tab_q.lungRiver = res$q 

col_name = c("River_Flu_lung")
colnames(tab_q.lungRiver) = col_name
tab_q.lungRiver
tab_q.lungRiver$ASV = rownames(tab_q.lungRiver)

FluDF = subset(tab_q.lungRiver, tab_q.lungRiver$River_Flu_lung < 0.05)

names = FluDF$ASV
FluTax = tax_table(ps.lung)[names,] %>%
  data.frame(.)
FluTax$ASV = rownames(FluTax)
FluDF = left_join(FluDF, FluTax, by = "ASV")
CtrlDFFlu = FluDF
```


```{r, ANCOM-BC for River fecal flu samples#######}
sample_data(ps.fecal)$Soil = factor(sample_data(ps.fecal)$Soil, levels = c("Ctrl", "Pine","River", "Road"))
sample_data(ps.fecal)$Flu = as.factor(sample_data(ps.fecal)$Flu)
sample_data(ps.fecal)$Soil
sample_data(ps.fecal)$Flu

ps.fecalRiver = subset_samples(ps.fecal, sample_data(ps.fecal)$Soil == "River" & sample_data(ps.fecal)$Time == "35") %>%
    prune_species(speciesSums(.) > 0, .)

out = ancombc(phyloseq = ps.fecalRiver, formula = "Flu", 
              p_adj_method = "BH", zero_cut = 0.95, lib_cut = 1000, 
              group = "Flu", struc_zero = F, neg_lb = F, tol = 1e-5, 
              max_iter = 100, conserve = TRUE, alpha = 0.05, global = TRUE)

out.lungRiver = out
res = out.lungRiver$res
tab_q.lungRiver = res$q 

col_name = c("River_Flu_lung")
colnames(tab_q.lungRiver) = col_name
tab_q.lungRiver
tab_q.lungRiver$ASV = rownames(tab_q.lungRiver)

FluDF = subset(tab_q.lungRiver, tab_q.lungRiver$River_Flu_lung < 0.05)

names = FluDF$ASV
FluTax = tax_table(ps.fecal)[names,] %>%
  data.frame(.)
FluTax$ASV = rownames(FluTax)
FluDF = left_join(FluDF, FluTax, by = "ASV")

riverDFecal = FluDF
sample_names(ps.fecalRiver)
```

```{r, ANCOM-BC for Road fecal flu samples}
sample_data(ps.fecal)$Soil = factor(sample_data(ps.fecal)$Soil, levels = c("Ctrl", "Pine","River", "Road"))
sample_data(ps.fecal)$Flu = as.factor(sample_data(ps.fecal)$Flu)
sample_data(ps.fecal)$Soil
sample_data(ps.fecal)$Flu

ps.fecalRoad = subset_samples(ps.fecal, sample_data(ps.fecal)$Soil == "Road" & sample_data(ps.fecal)$Time == "35") %>%
    prune_species(speciesSums(.) > 0, .)

out = ancombc(phyloseq = ps.fecalRoad, formula = "Flu", 
              p_adj_method = "BH", zero_cut = 0.95, lib_cut = 1000, 
              group = "Flu", struc_zero = F, neg_lb = F, tol = 1e-5, 
              max_iter = 100, conserve = TRUE, alpha = 0.05, global = TRUE)

out.lungRoad = out
res = out.lungRoad$res
tab_q.lungRoad = res$q 

col_name = c("Road_Flu_lung")
colnames(tab_q.lungRoad) = col_name
tab_q.lungRoad
tab_q.lungRoad$ASV = rownames(tab_q.lungRoad)

FluDF = subset(tab_q.lungRoad, tab_q.lungRoad$Road_Flu_lung < 0.05)

names = FluDF$ASV
FluTax = tax_table(ps.fecal)[names,] %>%
  data.frame(.)
FluTax$ASV = rownames(FluTax)
FluDF = left_join(FluDF, FluTax, by = "ASV")

RoadDFecal = FluDF
```

```{r, ANCOM-BC for Pine fecal flu samples}
sample_data(ps.fecal)$Soil = factor(sample_data(ps.fecal)$Soil, levels = c("Ctrl", "Pine","River", "Road"))
sample_data(ps.fecal)$Flu = as.factor(sample_data(ps.fecal)$Flu)
sample_data(ps.fecal)$Soil
sample_data(ps.fecal)$Flu

ps.fecalPine = subset_samples(ps.fecal, sample_data(ps.fecal)$Soil == "Pine" & sample_data(ps.fecal)$Time == "35") %>%
    prune_species(speciesSums(.) > 0, .)

out = ancombc(phyloseq = ps.fecalPine, formula = "Flu", 
              p_adj_method = "BH", zero_cut = 0.95, lib_cut = 1000, 
              group = "Flu", struc_zero = F, neg_lb = F, tol = 1e-5, 
              max_iter = 100, conserve = TRUE, alpha = 0.05, global = TRUE)

out.lungPine = out
res = out.lungPine$res
tab_q.lungPine = res$q 

col_name = c("Pine_Flu_lung")
colnames(tab_q.lungPine) = col_name
tab_q.lungPine
tab_q.lungPine$ASV = rownames(tab_q.lungPine)

FluDF = subset(tab_q.lungPine, tab_q.lungPine$Pine_Flu_lung < 0.05)

names = FluDF$ASV
FluTax = tax_table(ps.fecal)[names,] %>%
  data.frame(.)
FluTax$ASV = rownames(FluTax)
FluDF = left_join(FluDF, FluTax, by = "ASV")

PineDFecal = FluDF
sample_names(ps.fecalPine)
```

```{r, ANCOM-BC for Ctrl lung flu samples}
sample_data(ps.fecal)$Soil = factor(sample_data(ps.fecal)$Soil, levels = c("Ctrl", "Pine","River", "Road"))
sample_data(ps.fecal)$Flu = as.factor(sample_data(ps.fecal)$Flu)
sample_data(ps.fecal)$Soil
sample_data(ps.fecal)$Flu

ps.fecalCtrl = subset_samples(ps.fecal, sample_data(ps.fecal)$Soil == "Ctrl" & sample_data(ps.fecal)$Time == "35") %>%
    prune_species(speciesSums(.) > 0, .)

out = ancombc(phyloseq = ps.fecalCtrl, formula = "Flu", 
              p_adj_method = "BH", zero_cut = 0.95, lib_cut = 1000, 
              group = "Flu", struc_zero = F, neg_lb = F, tol = 1e-5, 
              max_iter = 100, conserve = TRUE, alpha = 0.05, global = TRUE)

sample_names(ps.fecalCtrl)

out.lungCtrl = out
res = out.lungCtrl$res
tab_q.lungCtrl = res$q 

col_name = c("Ctrl_Flu_lung")
colnames(tab_q.lungCtrl) = col_name
tab_q.lungCtrl
tab_q.lungCtrl$ASV = rownames(tab_q.lungCtrl)

FluDF = subset(tab_q.lungCtrl, tab_q.lungCtrl$Ctrl_Flu_lung < 0.05)

names = FluDF$ASV
FluTax = tax_table(ps.fecal)[names,] %>%
  data.frame(.)
FluTax$ASV = rownames(FluTax)
FluDF = left_join(FluDF, FluTax, by = "ASV")

CtrlDFecal = FluDF
```

```{r}
BiocManager::install("microbiome")
library(microbiome)

x = microbiome::transform(ps.fecal.35, "clr")

m = ordinate(x, method = "RDA", distance = "euclidean") %>%
  plot_ordination(x, ., type="samples", color = "Soil") +
  theme_minimal() +
  stat_ellipse() +
  ggtitle("PCoA of Day 35 Fecal Samples")


y= microbiome::transform(ps.lung.r, "clr")
  
g = ordinate(y, method = "RDA", distance = "euclidean") %>%
  plot_ordination(y, ., type="samples", color = "Soil") +
  theme_minimal() +
  stat_ellipse() +
  ggtitle("PCoA of Lung Samples")

r = ggarrange(m,g, labels = c("A", "B"), nrow = 2)


PermDF = data.frame(sample_data(y))
BetaD=phyloseq::distance(y, method = "euclidean") 

#permanova by soil
adonis(BetaD~Soil*Flu, permutations = 999, data=PermDF)
```

```{r}
y = microbiome::transform(ps.fecal.C, "clr")

ordinate(y, method = "RDA", distance = "euclidean") %>%
  plot_ordination(y, ., type="samples", color = "Soil") +
  theme_minimal() +
  stat_ellipse() +
  ggtitle("PCoA of Lung Samples")

PermDF = data.frame(sample_data(y))
BetaD=phyloseq::distance(y, method = "euclidean") 

#permanova by soil
adonis(BetaD~Soil*Flu, permutations = 999, data=PermDF)
```

```{r}
x = refseq(ps.fecal.C) %>%
  data.frame(.)

#calls sequence and abundance columns 
x$sequence = x$.

x$abundance = "1"

uniques=dada2::getUniques(x)

sequences=rownames(x)

dada2::uniquesToFasta(uniques,fout = "~/mouseenv/fecalseq.fasta", ids=sequences)

otu = as.data.frame(cbind(t(otu_table(ps.fecal.C, taxa_are_rows = FALSE)), tax_table(ps.fecal.C)))

write.table(otu, "~/mouseenv/fecalOTU.tsv", sep = "\t")
```


```{r}
MouseLungMetaData = read.csv("~/mouseenv/MouseLungMetaData.csv")

FamDFASV$bind = paste0(FamDFASV$Soil,FamDFASV$Flu,FamDFASV$Ear,FamDFASV$Cohort)

MouseLungMetaData = MouseLungMetaData %>%
  rename(., Control = Ctrl)

MouseLungMetaData = MouseLungMetaData %>%
mutate(Soil = str_replace(Soil, "Control", "Ctrl"))

corDF = merge(FamDFASV, MouseLungMetaData, by = c("Soil","Cohort","Flu", "Ear"))

ggplot(corDF, aes(x = X51121, y = Neutrophils, color = Soil)) + 
  geom_point() + 
  geom_smooth(method = "lm") +
  theme_minimal() + 
  ylab("Cage Shannon diversity")+ 
  xlab("Inverse Lung Shannon diversity") +
  facet_grid(~Flu)

ggscatter(corDFPR8, x = "logShannon", y = "Shannon.y", 
          add = "reg.line", conf.int = TRUE, 
          cor.coef = TRUE, cor.method = "pearson",
          xlab = "Lung", ylab = "Cage")
```

```{r}
library(biomformat);packageVersion("biomformat")
## [1] 1.6.0

otub<-t(as(otu_table(ps.lung),"matrix")) # 't' to transform if taxa_are_rows=FALSE
#if taxa_are_rows=TRUE
#otu<-as(otu_table(GlobalPatterns),"matrix"))
otu_biom<-make_biom(data=otub)
write_biom(otu_biom,"~/otu_biom.biom")

ig <- make_network(ps.soil, "samples", max.dist=0.8)

plot_network(ig, ps.soil, color="Soil",line_weight=0.3, label=NULL)
```


#can likely remove
```{r}
lungDF = lungDF %>%
  data.frame(.)

metaDFb = read.csv("~/mouseenv/MouseLungMetaData.csv") %>%
  rename(., "Virus" = "Flu")

corDF = merge(lungDF, metaDFb, by = c("Soil","Cohort","Flu", "Ear"))

rownames(corDF) = corDF$SampleName
list = rownames(corDF)
shannonb = Shannon.result[list,]

cordDFb = cbind(corDF, shannonb$Shannon)

cordDFbPR8 = subset(cordDFb, cordDFb$Flu == "PR8")

cordDFbHI = subset(cordDFb, cordDFb$Flu == "HI")

ggplot(cordDFbPR8, aes(x = Neutrophils, y = shannonb$Shannon)) + 
  geom_point()

ggplot(cordDFbHI, aes(x = Neutrophils, y = shannonb$Shannon)) + 
  geom_point() 
```

```{r, do fecal taxa show up later in Day 14 or 35 soils? Pine first}
ps.soil.pine = subset_samples(ps.soil.r, sample_data(ps.soil.r)$Soil == "Road")

otu <- otu_table(ps.soil.pine)

bad_taxa <- ps.soil.pine %>%
    subset_samples(Time == "0") %>%
    prune_species(speciesSums(.) > 0, .) %>%
    taxa_sums %>%
    sort(decreasing = TRUE) %>%
    names

names = sample_names(ps.soil.pine)

otu[names, bad_taxa] <- 0

otu_table(ps.soil.pine) <- otu

sample_sums(ps.soil.pine)

ps.soil.f = prune_species(speciesSums(ps.soil.pine) > 0, ps.soil.pine)

#need to remove bad taxa from fecal samples
ps.fecal.0  = subset_samples(ps.fecal.r, sample_data(ps.fecal.r)$Time == "0")

otu = otu_table(ps.fecal.0)

names = sample_names(ps.fecal.0)

bad_taxa2 <- ps.fecal.0 %>%
    prune_species(speciesSums(.) > 0, .) %>%
    taxa_sums %>%
    sort(decreasing = TRUE) %>%
    names

badT = intersect(bad_taxa, bad_taxa2)

otu[names, badT] <- 0

otu_table(ps.fecal.0) <- otu

ps.fecal.f = prune_species(speciesSums(ps.fecal.0) > 0, ps.fecal.0)

#left join OTU tables?
good_taxa = ps.fecal.f %>%
    prune_species(speciesSums(.) > 0, .) %>%
    taxa_sums %>%
    sort(decreasing = TRUE) %>%
    names

good_taxa2 = ps.soil.f %>%
    prune_species(speciesSums(.) > 0, .) %>%
    taxa_sums %>%
    sort(decreasing = TRUE) %>%
    names

finalT = setdiff(good_taxa2, good_taxa)

otu = otu_table(ps.soil.f) 

names = sample_names(ps.soil.f)

otu[names, finalT] <- 0

otu_table(ps.soil.f) <- otu

ps.soil.f %>% subset_samples(., sample_data(ps.soil.f)$Time != "0") %>%
              sample_sums(.) %>%
              sd(.)

ps.soil.f %>% subset_samples(., sample_data(ps.soil.f)$Time != "0") %>%
              sample_sums(.) %>%
              mean(.)
```