---
title: "16SAnalysis"
output: html_document
---

```{r}
library(dada2)
library(dplyr)
library(stringr)
library(phyloseq)
#BiocManager::install("decontam")
library(decontam)
#BiocManager::install("ggtree")
library(ggtree)
library(ape)
#install.packages("picante")
library(picante)
#install.packages("caret")
library(caret)
#install.packages("MLmetrics")
library(MLmetrics)
install.packages("remotes")
remotes::install_github("Russel88/MicEco")
require(MicEco)
install.packages("gjam")
library(gjam)
remotes::install_github('reptalex/phylofactor')
library(phylofactor)
#BiocManager::install("ANCOMBC")
library(ANCOMBC)
install.packages("ggpubr")
library(ggpubr)
library(rstatix)
```

```{r}
# Directories
fastq.dir = file.path("/home/data/16Sfiles")
trimmed.dir = file.path("/home/data/trimmed")

# Set variables for bash
Sys.setenv(FASTQ_DIR = fastq.dir)
Sys.setenv(TRIMMED_DIR = trimmed.dir)
```

```{bash}
mkdir $TRIMMED_DIR

for FASTQ in $FASTQ_DIR/*_R1_001.fastq.gz
  do
    FASTQ_BASE="$(basename ${FASTQ} '_001.fastq.gz')"
    echo "---------------- TRIMMING: $FASTQ ----------------"
    $HOME/.local/bin/cutadapt -g XTCGCATAGTG -o $TRIMMED_DIR/${FASTQ_BASE}.trim.fastq.gz $FASTQ >>/home/data/cutadaptF.txt
    done

for FASTQ in $FASTQ_DIR/*_R2_001.fastq.gz
  do
    FASTQ_BASE="$(basename ${FASTQ} '_001.fastq.gz')"
    echo "---------------- TRIMMING: $FASTQ ----------------"
    $HOME/.local/bin/cutadapt -g XTCGCATAGGACTAC -o $TRIMMED_DIR/${FASTQ_BASE}.trim.fastq.gz $FASTQ >>/home/data/cutadaptR.txt
    done
```

```{r}
path <- "/home/data/trimmed" 
list.files(path)
```

```{r}
# Forward and reverse fastq filenames have format: SAMPLENAME_R1_001.fastq and SAMPLENAME_R2_001.fastq
fnFs <- sort(list.files(path, pattern="_R1.trim.fastq.gz", full.names = TRUE))
fnRs <- sort(list.files(path, pattern="_R2.trim.fastq.gz", full.names = TRUE))
# Extract sample names, assuming filenames have format: SAMPLENAME_XXX.fastq
sample.names <- str_remove(basename(fnFs), "_L002_R1.trim.fastq.gz")

```

```{r}
# Place filtered files in filtered/ subdirectory
filtFs <- file.path(path, "filtered", paste0(sample.names, "_F_filt.fastq.gz"))
filtRs <- file.path(path, "filtered", paste0(sample.names, "_R_filt.fastq.gz"))
names(filtFs) <- sample.names
names(filtRs) <- sample.names

out <- filterAndTrim(fnFs, filtFs, fnRs, filtRs, truncLen=c(230,190),
              maxN=0, maxEE=c(3,3), trimLeft = c(16,13), truncQ=2, rm.phix=TRUE,
              compress=TRUE, multithread=F) 

errF <- learnErrors(filtFs, nbases = 1e10, multithread=TRUE)
errR <- learnErrors(filtRs, nbases = 1e10, multithread=TRUE)
plotErrors(errF, nominalQ=T)
```

```{r}

dadaFs <- dada(filtFs[1:20], err=errF, multithread=TRUE)
dadaRs <- dada(filtRs[1:20], err=errR, multithread=TRUE)

mergers <- mergePairs(dadaFs, filtFs[1:20], dadaRs, filtRs[1:20], verbose=TRUE)
seqtab <- makeSequenceTable(mergers)

seqtab2 <- seqtab[,nchar(colnames(seqtab)) %in% 248:260]

seqtab.nochim <- removeBimeraDenovo(seqtab2, method="consensus", multithread=TRUE, verbose=TRUE)

table(nchar(getSequences(seqtab.nochim)))

sum(seqtab.nochim)/sum(seqtab)

track2 <- cbind(out[1:20], sapply(dadaFs, getN), sapply(dadaRs, getN), sapply(mergers, getN), rowSums(seqtab.nochim))
# If processing a single sample, remove the sapply calls: e.g. replace sapply(dadaFs, getN) with getN(dadaFs)
colnames(track) <- c("input", "filtered", "denoisedF", "denoisedR", "merged", "nonchim")
rownames(track) <- sample.names
```


```{r}
newFs <- list.files("/home/data/trimmed/filtered", pattern="_F_filt.fastq.gz")

sample.names = str_remove(basename(newFs), "_F_filt.fastq.gz")

mergers <- vector("list", length(sample.names))
names(mergers) <- sample.names
for(sam in sample.names) {
  cat("Processing:", sam, "\n")
    derepF <- derepFastq(filtFs[[sam]])
    ddF <- dada(derepF, err=errF, multithread=TRUE)
    derepR <- derepFastq(filtRs[[sam]])
    ddR <- dada(derepR, err=errR, multithread=TRUE)
    merger <- mergePairs(ddF, derepF, ddR, derepR)
    mergers[[sam]] <- merger
}

rm(derepF); rm(derepR)

seqtab <- makeSequenceTable(mergers)

seqtab2 <- seqtab[,nchar(colnames(seqtab)) %in% 251:255]

rm(seqtab)

saveRDS(seqtab2, "~/seqtab.RDS")

table(nchar(getSequences(seqtab2)))

seqtab.nochim <- removeBimeraDenovo(seqtab2, method="consensus", multithread=TRUE, verbose=TRUE)

saveRDS(seqtab.nochim, "~/seqtab.nochim.RDS")
```

```{r}
getN <- function(x) sum(getUniques(x))
track <- cbind(sapply(dadaFs, getN), sapply(dadaRs, getN), sapply(mergers, getN), rowSums(seqtab.nochim))
# If processing a single sample, remove the sapply calls: e.g. replace sapply(dadaFs, getN) with getN(dadaFs)
colnames(track) <- c("denoisedF", "denoisedR", "merged", "nonchim")
rownames(track) <- sample.names
track =  data.frame(track)
head(track)
```

```{bash, include = F}
wget http://www2.decipher.codes/Classification/TrainingSets/SILVA_SSU_r138_2019.RData -P ./silvadb
```

```{r}
library(DECIPHER)
dna <- DNAStringSet(getSequences(seqtab.nochim)) # Create a DNAStringSet from the ASVs

load("~/silvadb/SILVA_SSU_r138_2019.RData")

ids <- IdTaxa(dna, trainingSet, strand="top", processors=30, verbose=FALSE)

ranks <- c("domain", "phylum", "class", "order", "family", "genus") # ranks of interest

# Convert the output object of class "Taxa" to a matrix analogous to the output from assignTaxonomy
taxid <- t(sapply(ids, function(x) {
        m <- match(ranks, x$rank)
        taxa <- x$taxon[m]
        taxa[startsWith(taxa, "unclassified_")] <- NA
        taxa
}))
colnames(taxid) <- ranks; rownames(taxid) <- getSequences(seqtab.nochim)

saveRDS(taxid, "~/taxid.RDS")
```

```{r}
seqtab.nochim = readRDS("~/seqtab.nochim.RDS")
taxid = readRDS("~/taxid.RDS")
sampleDF = read.csv("~/mouseenv/mouse16SmetaD.csv")

rownames(sampleDF) = sampleDF$SampleName
```

```{r}
colSums(t(seqtab.nochim) != 0)
rowSums(seqtab.nochim)
```

##Creation of tree
```{bash, make a tree with SEPP using gg backbone tree}
wget  "https://raw.github.com/smirarab/sepp-refs/master/gg/sepp-package.tar.bz"

tar xvfj sepp-package.tar.bz

cd sepp-package/sepp

python3 setup.py config -c
```

```{r, make list of sequences in fasta file for tree making}
uniques=getUniques(seqtab.nochim)

sequences=names(uniques)

uniquesToFasta(uniques,fout = "~/mouseenv/seqs.fasta", ids=sequences)
```

```{bash}
./sepp-package/run-sepp.sh sepp-package/test.frag test-gg
```

```{bash}
./sepp-package/run-sepp.sh ./seqs.fasta MOUSE
```

```{r, view tree file with ggtree}
seq.tree = read.tree(file = "~/mouseenv/MOUSE_placement.tog.relabelled.tre") %>%
  keep.tip(., sequences)
```

```{r}
ps <- phyloseq(otu_table(seqtab.nochim, taxa_are_rows=FALSE), 
               sample_data(sampleDF), 
               tax_table(taxid),
              phy_tree(seq.tree))

saveRDS(ps, "~/mouseenv/ps.RDS")

ps = readRDS("~/mouseenv/ps.RDS")
```

```{r}
vegan_otu <- function(physeq) {
    OTU <- otu_table(physeq)
    if (taxa_are_rows(OTU)) {
        OTU <- t(OTU)
    }
    return(as(OTU, "matrix"))
}
```

```{r, remove contaminating sequences}
ps.Bact=subset_taxa(ps, domain=="Bacteria") %>%
  prune_species(speciesSums(.) > 0, .) 

library(ggplot2)

neg = ifelse(sample_data(ps.Bact)$SampleType == "Blank", T, F)

contamdf.prev <- isContaminant(ps.Bact, method="prevalence", neg=neg, threshold = 0.5)

table(contamdf.prev$contaminant)

hist(contamdf.prev$p, 100, ylim = c(0,5000), xlim = c(0,1))

ps.noncontam = prune_taxa(!contamdf.prev$contaminant, ps.Bact)

ps.final = subset_samples(ps.noncontam, sample_data(ps)$SampleType != "Blank")%>%
  prune_species(speciesSums(.) > 0, .) 

min(sample_sums(ps.final))

# Number ASVs and store the exact sequences in the reference sequence slot - can access with refseq(ps)
dna <- Biostrings::DNAStringSet(taxa_names(ps.final))
names(dna) <- taxa_names(ps.final)
ps.final <- merge_phyloseq(ps.final, dna)
taxa_names(ps.final) <- paste0(seq(ntaxa(ps.final)))
ps.final
remove(dna)
rm(ps.Bact,ps.noncontam,neg,ps, contamdf.prev)

saveRDS(ps.final, "~/mouseenv/ps_final.RDS")
```

```{r, environment-lung connection}
ps.lung = subset_samples(ps.final, sample_data(ps.final)$SampleType == "Lung" & sample_data(ps.final)$Rep == "A") %>%
  subset_taxa(., phylum != "NA") %>%
  prune_species(speciesSums(.) > 0, .) 

table(tax_table(ps.lung)[, "phylum"], exclude = NULL)

min(sample_sums(ps.lung))

ps.lung.r = transform_sample_counts(ps.lung, function(x) x/sum(x)) %>%
  prune_species(speciesSums(.) > 0, .)

set.seed(42)
ordinate(ps.lung.r, method = "NMDS", distance = "bray", trymax = 1000) %>%
  plot_ordination(ps.lung.r, ., type="samples", color = "Soil") +
  theme_minimal() +
  stat_ellipse() +
  facet_grid(~Flu) 

ordinate(ps.lung.r, method = "PCoA", distance = "bray") %>%
  plot_ordination(ps.lung.r, ., type="samples", color = "Soil") +
  theme_minimal() +
  stat_ellipse() +
  ggtitle("Lung HI microbiome, PCoA, Bray-Curtis")

PermDF = data.frame(sample_data(ps.lung.r))
BetaD=vegdist(vegan_otu(ps.lung.r), "bray")

#permanova by soil
adonis(BetaD~Soil*Flu, permutations = 999, data=PermDF)

#Shannon diversity 
Shannon.result = estimate_richness(ps.lung.r, measures = "Shannon")

Shannon.result$Samples = rownames(Shannon.result)
lungDF = sample_data(ps.lung.r)
Shannon = cbind(lungDF, Shannon.result)

hist(Shannon$Shannon)
ggqqplot(Shannon$Shannon)


Shannon$logShannon = log10(max(Shannon$Shannon+1) - Shannon$Shannon)

shapiro.test(Shannon$Shannon)

shapiro.test(Shannon$logShannon)

hist(Shannon$Shannon)
ggqqplot(Shannon$Shannon)
hist(Shannon$logShannon)
ggqqplot(Shannon$logShannon)

fit = aov(logShannon ~ Soil*Flu, data = Shannon)
summary(fit)

TukeyHSD(fit)

p = ggplot(Shannon, aes(Soil, logShannon, fill = Soil)) + 
  geom_boxplot() +
  expand_limits(y=0)  + 
  theme_minimal() +
  xlab(element_blank())+ 
  facet_grid(~Flu) + 
  ylab("log(max(Diversity + 1) - Diversity") 

#############

Observed.result = estimate_richness(ps.lung, measures = "Observed")

Observed.result$Samples = rownames(Observed.result)

lungDF = sample_data(ps.lung)

Observed = cbind(lungDF, Observed.result)

g=ggplot(Observed, aes(Soil, Observed, fill = Soil)) + 
  geom_boxplot() +
  expand_limits(y=0)  + 
  theme_minimal() +
  xlab(element_blank())+ 
  facet_grid(~Flu) + 
  ylab("Richness")

r = ggarrange(p,g, labels = c("A", "B"), nrow = 2)

hist(Observed$Observed)
ggqqplot(Observed$Observed)

shapiro.test(Observed$Observed)

fit = aov(Observed ~ Soil*Flu, data = Observed)
summary(fit)

TukeyHSD(fit)
```

#can likely remove
```{r}
lungDF = lungDF %>%
  data.frame(.)

metaDFb = read.csv("~/mouseenv/MouseLungMetaData.csv") %>%
  rename(., "Virus" = "Flu")

corDF = merge(lungDF, metaDFb, by = c("Soil","Cohort","Flu", "Ear"))

rownames(corDF) = corDF$SampleName
list = rownames(corDF)
shannonb = Shannon.result[list,]

cordDFb = cbind(corDF, shannonb$Shannon)

cordDFbPR8 = subset(cordDFb, cordDFb$Flu == "PR8")

cordDFbHI = subset(cordDFb, cordDFb$Flu == "HI")

ggplot(cordDFbPR8, aes(x = Neutrophils, y = shannonb$Shannon)) + 
  geom_point()

ggplot(cordDFbHI, aes(x = Neutrophils, y = shannonb$Shannon)) + 
  geom_point() 
```

```{r, soil day 0}
#changed from ps.final
ps.soil = subset_samples(ps.final, sample_data(ps.final)$SampleType == "Soil") %>%
  prune_species(speciesSums(.) > 0, .) 

ps.soil.r = transform_sample_counts(ps.soil, function(x) x/sum(x)) %>%
  prune_species(speciesSums(.) > 0, .)

ps.soil.0 = subset_samples(ps.final, sample_data(ps.final)$SampleType == "Soil" & sample_data(ps.final)$Time == "0") %>%
  prune_species(speciesSums(.) > 0, .) 

#Shannon diversity
Shannon.result = estimate_richness(ps.soil.0, measures = "Shannon")

Shannon.result$Samples = rownames(Shannon.result)
soilDF = sample_data(ps.soil.0)
Shannon = cbind(soilDF, Shannon.result) 

shapiro.test(Shannon$Shannon)
hist(Shannon$Shannon)
Shannon$logShannon = log(Shannon$Shannon)
hist(Shannon$logShannon)

shapiro.test(Shannon$logShannon)

kruskal.test(Shannon ~ Soil, data = Shannon)

ggqqplot(Shannon$Shannon)

fit = aov(Shannon ~ Soil, data = Shannon)
summary(fit)

TukeyHSD(fit)

p=ggplot(Shannon, aes(Soil, Shannon, fill = Soil)) + 
  geom_boxplot() +
  expand_limits(y=0)  + 
  theme_minimal() +
  xlab(element_blank())+
  scale_fill_manual(values = c("#7CAE00", "#00BFC4", "#C77CFF")) +
  ylab("Shannon Diversity")+ 
  theme(legend.position = "none")

#Observed richness 
Observed.result = estimate_richness(ps.soil.0, measures = "Observed")

Observed.result$Samples = rownames(Observed.result)
soilDF = sample_data(ps.soil.0)
Observed = cbind(soilDF, Observed.result) 

g=ggplot(Observed, aes(Soil, Observed, fill = Soil)) + 
  geom_boxplot() +
  expand_limits(y=0)  + 
  theme_minimal() +
  ylab("Richness") +
  scale_fill_manual(values = c("#7CAE00", "#00BFC4", "#C77CFF")) +
  xlab(element_blank())+ 
  theme(legend.position = "none")

shapiro.test(Observed$Observed)
hist(Observed$Observed)

kruskal.test(Observed ~ Soil, data = Observed)

fit = aov(Observed ~ Soil, data = Observed)
summary(fit)
plot(fit)

TukeyHSD(fit)

#######################################

r=ordinate(ps.soil.0, method = "NMDS", distance = "bray", trymax = 1000) %>%
  plot_ordination(ps.soil.0, ., type="samples", color = "Soil") +
  theme_minimal() +
  scale_color_manual(values = c("#7CAE00", "#00BFC4", "#C77CFF")) +
  stat_ellipse()

PermDF = data.frame(sample_data(ps.soil.0))
BetaD=vegdist(vegan_otu(ps.soil.0), "bray")

#permanova by soil
adonis(BetaD~Soil, permutations = 999, data=PermDF)

m = ggarrange(r, 
              ggarrange(p,g, ncol = 2, labels = c("B", "C")), nrow = 2, labels = "A")
```

```{r, day 14 and 35 soils}
ps.soil.C = subset_samples(ps.final, sample_data(ps.final)$SampleType == "Soil" & sample_data(ps.final)$Time != "0") %>%
  prune_species(speciesSums(.) > 0, .) 

Shannon.result = estimate_richness(ps.soil.C, measures = "Shannon")

Shannon.result$Samples = rownames(Shannon.result)
soilDF = sample_data(ps.soil.C)
Shannon = cbind(soilDF, Shannon.result) 

summary(Shannon$Shannon)

shapiro.test(Shannon$Shannon)

hist(Shannon$Shannon)

Shannon$logShannon = log(Shannon$Shannon)
hist(Shannon$logShannon)

ggqqplot(Shannon$Shannon)

fit = aov(Shannon ~ Soil, data = Shannon)
summary(fit)

TukeyHSD(fit)

p=ggplot(Shannon, aes(Soil, Shannon, fill = Soil)) + 
  geom_boxplot() +
  expand_limits(y=0)  + 
  theme_minimal() +
  xlab(element_blank())+
  ylab("Shannon Diversity")+ 
  theme(legend.position = "none")

#Observed richness 
Observed.result = estimate_richness(ps.soil.C, measures = "Observed")

Observed.result$Samples = rownames(Observed.result)
soilDF = sample_data(ps.soil.C)
Observed = cbind(soilDF, Observed.result) 

hist(Observed$Observed)

kruskal.test(Observed ~ Soil, Observed)

Observed %>%
  pairwise_wilcox_test(Observed ~ Soil, p.adjust.methods = "BH")

g=ggplot(Observed, aes(Soil, Observed, fill = Soil)) + 
  geom_boxplot() +
  expand_limits(y=0)  + 
  theme_minimal() +
  ylab("Richness") +
  xlab(element_blank())+ 
  theme(legend.position = "none")

#######################################

r=ordinate(ps.soil.C, method = "NMDS", distance = "bray", trymax = 1000) %>%
  plot_ordination(ps.soil, ., type="samples", color = "Soil") +
  theme_minimal() +
  stat_ellipse() 

PermDF = data.frame(sample_data(ps.soil.C))
BetaD=vegdist(vegan_otu(ps.soil.C), "bray")

#permanova by soil
adonis(BetaD~Soil, permutations = 999, data=PermDF)

m = ggarrange(r, 
              ggarrange(p,g, ncol = 2, labels = c("B", "C")), nrow = 2, labels = "A")
```

```{r}
PermDF = subset_samples(ps.soil.r, sample_data(ps.soil.r)$Time != "0") %>%
  sample_data(.) %>%
  data.frame(.)

BetaD = subset_samples(ps.soil.r, sample_data(ps.soil.r)$Time != "0") %>%
  vegan_otu(.) %>%
  vegdist(., "bray")

#permanova by soil
adonis(BetaD~Soil*Flu*Time, permutations = 999, data=PermDF)

#####################
Shannon.result = estimate_richness(ps.soil, measures = "Shannon")

Shannon.result$Samples = rownames(Shannon.result)

Shannon = cbind(soilDF, Shannon.result)

ggplot(Shannon, aes(Time, Shannon, fill = Soil)) + 
  geom_boxplot() +expand_limits(y=0)  + 
  theme_minimal() +
  #scale_fill_manual(values = c("#BF812D","darkslategray4"))+
  xlab(element_blank())+ 
  facet_grid(~Soil)

sample = vegan_otu(ps.soil)
tree = phy_tree(ps.soil)
pd.result = pd(sample, tree)

FPD = cbind(pd.result, soilDF) 

FPD$Time = as.factor(FPD$Time)

ggplot(FPD, aes(Time, PD, fill = Soil)) + 
  geom_boxplot() +
  expand_limits(y=0)  + 
  theme_minimal() +
  #scale_fill_manual(values = c("#BF812D","darkslategray4"))+
  xlab(element_blank())+ 
  facet_grid(~Soil)

ShannonD0 = subset(Shannon, Time < 1)

ggplot(ShannonD0, aes(Soil, Shannon, fill = Soil)) + 
  geom_boxplot() +expand_limits(y=0)  + 
  theme_minimal() +
  #scale_fill_manual(values = c("#BF812D","darkslategray4"))+
  xlab(element_blank())
```

```{r, correlation with cage-lung}
#######
Shannon.result = estimate_richness(ps.lung, measures = "Shannon")

Shannon.result$Samples = rownames(Shannon.lung)
lungDF = sample_data(ps.lung)
Shannon.lung = cbind(lungDF, Shannon.result)

Shannon.lung$logShannon = log10(max(Shannon.lung$Shannon+1) - Shannon.lung$Shannon)

Shannon.lung$join = paste(Shannon.lung$Soil, Shannon.lung$Cohort, Shannon.lung$Flu)
#######
Shannon.result = estimate_richness(ps.soil, measures = "Shannon")

Shannon.result$Samples = rownames(Shannon.result)
soilDF = sample_data(ps.soil.r)

Shannon.soil = cbind(soilDF, Shannon.result) %>%
  subset(., Time == "35")

Shannon.soil$join = paste(Shannon.soil$Soil, Shannon.soil$Cohort, Shannon.soil$Flu)

Shannon.soil = select(Shannon.soil, Shannon, join)

corDFHI = merge(Shannon.lung, Shannon.soil, by = "join") %>%
  subset(., Flu == "HI")

p = ggplot(corDF, aes(x = logShannon, y = Shannon.y)) + 
  geom_point() + 
  geom_smooth(method = "lm") +
  theme_minimal() + 
  ylab("Cage Shannon diversity")+ 
  xlab("Inverse Lung Shannon diversity")

corDFPR8 = merge(Shannon.lung, Shannon.soil, by = "join") %>%
  subset(., Flu == "PR8")

corDF = rbind(corDFHI, corDFPR8)

g=ggplot(corDF, aes(x = logShannon, y = Shannon.y)) + 
  geom_point() + 
  geom_smooth(method = "lm") +
  theme_minimal() + 
  ylab("Cage Shannon diversity")+ 
  xlab("Inverse Lung Shannon diversity") +
  facet_grid(~Flu)

ggscatter(corDFPR8, x = "logShannon", y = "Shannon.y", 
          add = "reg.line", conf.int = TRUE, 
          cor.coef = TRUE, cor.method = "pearson",
          xlab = "Lung", ylab = "Cage")
```

```{r}
#changed from ps.final
ps.fecal = subset_samples(ps.final, sample_data(ps.final)$SampleType == "Fecal") %>%
  subset_taxa(., phylum != "NA") %>%
  prune_species(speciesSums(.) > 0, .) 

table(tax_table(ps.fecal)[, "phylum"], exclude = NULL)

min(sample_sums(ps.fecal))

ps.fecal.r = transform_sample_counts(ps.fecal, function(x) x/sum(x)) %>%
  prune_species(speciesSums(.) > 0, .)

fecalDF = sample_data(ps.fecal.r)

sample_data(ps.fecal.r)$Time = as.factor(fecalDF$Time)

set.seed(42)

p=ordinate(ps.fecal.r, method = "NMDS", distance = "bray", trymax = 1000) %>%
  plot_ordination(ps.fecal, ., type="samples", color = "Soil") +
  theme_minimal() +
  stat_ellipse()+ 
  facet_grid(~Time)

PermDF = data.frame(sample_data(ps.fecal.r))
BetaD=vegdist(vegan_otu(ps.fecal.r), "bray")

#permanova by soil
adonis(BetaD~Soil*Time, permutations = 999, data=PermDF)

PermDF = subset_samples(ps.fecal.r, sample_data(ps.fecal.r)$Time == "35") %>%
  sample_data(.) %>%
  data.frame(.)

BetaD=subset_samples(ps.fecal.r, sample_data(ps.fecal.r)$Time == "35") %>%
  vegan_otu(.) %>%
  vegdist(., "bray")
#permanova by soil
adonis(BetaD~Soil*Flu, permutations = 999, data=PermDF)

##################################
Shannon.result = estimate_richness(ps.fecal.r, measures = "Shannon")

Shannon.result$Samples = rownames(Shannon.result)

fecalDF = sample_data(ps.fecal.r)

Shannon = cbind(fecalDF, Shannon.result)

Shannon$Time = as.factor(Shannon$Time)

shapiro.test(Shannon$Shannon)
hist(Shannon$Shannon)
Shannon$logShannon = log10(max(Shannon$Shannon+1) - Shannon$Shannon)
shapiro.test(Shannon$logShannon)
hist(Shannon$logShannon)

fit = aov(logShannon ~ Soil*Time, data = Shannon)
summary(fit)

TukeyHSD(fit)

p=ggplot(Shannon, aes(Time, logShannon, fill = Soil)) + 
  geom_boxplot() +
  expand_limits(y=0)  + 
  theme_minimal() +
  xlab(element_blank())+ 
  facet_grid(~Soil) +
  ylab("log(max(Diversity + 1) - Diversity") + 
  xlab("Time (days)")+
  scale_y_continuous(limits = c(0,0.6)) + 
  theme(legend.position = "none")

####################

Observed.result = estimate_richness(ps.fecal, measures = "Observed")

Observed.result$Samples = rownames(Observed.result)

fecalDF = sample_data(ps.fecal)

Observed = cbind(fecalDF, Observed.result)

Observed$Time = as.factor(Observed$Time)

hist(Observed$Observed)
ggqqplot(Observed$Observed)
shapiro.test(Observed$Observed)

Observed$SQObserved = sqrt(Observed$Observed)
hist(Observed$SQObserved)
ggqqplot(Observed$SQObserved)

g=ggplot(Observed, aes(Time, SQObserved, fill = Soil)) + 
  geom_boxplot() +
  expand_limits(y=0)  + 
  theme_minimal() +
  xlab(element_blank())+ 
  facet_grid(~Soil) +
  ylab("Square Root(Richness)") + 
  xlab("Time (days)")+
  scale_y_continuous(limits = c(0,30)) + 
  theme(legend.position = "none")

fit = aov(SQObserved ~ Soil*Time, data = Observed)
summary(fit)
TukeyHSD(fit)
plot(fit)

r = ggarrange(p,g, labels = c("A", "B"), nrow = 2)
```

```{r}
ps.FW = subset_samples(ps.final, sample_data(ps.final)$SampleType == "Water" | sample_data(ps.final)$SampleType == "Food") %>%
  prune_species(speciesSums(.) > 0, .) 

ordinate(ps.FW, method = "NMDS", distance = "bray", trymax = 1000) %>%
  plot_ordination(ps.FW, ., type="samples", color = "SampleType") +
  theme_classic() +
  stat_ellipse()

Shannon.result = estimate_richness(ps.FW, measures = "Shannon")

Shannon.result$Samples = rownames(Shannon.result)
lungDF = sample_data(ps.lung)
Shannon = cbind(lungDF, Shannon.result)

ggplot(Shannon, aes(SampleType, Shannon, fill = Soil)) + 
  geom_boxplot() +expand_limits(y=0)  + 
  theme_minimal() +
  #scale_fill_manual(values = c("#BF812D","darkslategray4"))+
  xlab(element_blank())+ 
  facet_grid(~Flu)
```

```{r, ANCOMBC of the lung}
sample_data(ps.lung)$Soil = as.factor(sample_data(ps.lung)$Soil)
sample_data(ps.lung)$Flu = as.factor(sample_data(ps.lung)$Flu)
sample_data(ps.lung)$Soil
sample_data(ps.lung)$Flu

out = ancombc(phyloseq = ps.lung, formula = "Soil + Flu", 
              p_adj_method = "BH", zero_cut = 0.95, lib_cut = 1000, 
              group = "Soil", struc_zero = F, neg_lb = F, tol = 1e-5, 
              max_iter = 100, conserve = TRUE, alpha = 0.05, global = TRUE)

out.lung = out
res = out.lung$res
tab_q.lung = res$q

col_name = c("Ctrl-Pine_lung", "Ctrl-River_lung", "Ctrl-Road_lung", "HI-PR8_lung")
colnames(tab_q.lung) = col_name
tab_q.lung
tab_q.lung$ASV = rownames(tab_q.lung)
```

```{r}
sample_data(ps.soil)$Soil = as.factor(sample_data(ps.soil)$Soil)
sample_data(ps.soil)$Flu = as.factor(sample_data(ps.soil)$Flu)
sample_data(ps.soil)$Soil
sample_data(ps.soil)$Flu

ps.soil.35 = subset_samples(ps.soil, sample_data(ps.soil)$Time == "35") %>%
  prune_species(speciesSums(.) > 0, .)

out = ancombc(phyloseq = ps.soil.35, formula = "Soil + Flu", 
              p_adj_method = "BH", zero_cut = 0.95, lib_cut = 1000, 
              group = "Soil", struc_zero = T, neg_lb = T, tol = 1e-5, 
              max_iter = 100, conserve = TRUE, alpha = 0.05, global = TRUE)

out.soil = out
res = out.soil$res
tab_q.soil = res$q

col_name = c("Ctrl-Pine_soil", "Ctrl-River_soil", "Ctrl-Road_soil", "HI-PR8_soil")
colnames(tab_q.soil) = col_name
tab_q.soil$ASV = rownames(tab_q.soil)
tab_q.soil

out = ancombc(phyloseq = ps.soil.35, formula = "Soil + Flu", 
              p_adj_method = "BH", zero_cut = 0.95, lib_cut = 1000, 
              group = "Soil", struc_zero = F, neg_lb = F, tol = 1e-5, 
              max_iter = 100, conserve = TRUE, alpha = 0.05, global = TRUE)

out.soil.strucZ = out
res = out.soil.strucZ$res
tab_q.soil.strucZ = res$q

col_name = c("Ctrl-Pine_soil", "Ctrl-River_soil", "Ctrl-Road_soil", "HI-PR8_soil")
colnames(tab_q.soil.strucZ) = col_name
tab_q.soil.strucZ$ASV = rownames(tab_q.soil.strucZ)
tab_q.soil.strucZ
```

```{r}
sample_data(ps.fecal)$Soil = as.factor(sample_data(ps.fecal)$Soil)
sample_data(ps.fecal)$Flu = as.factor(sample_data(ps.fecal)$Flu)
sample_data(ps.fecal)$Soil
sample_data(ps.fecal)$Flu

ps.fecal.35 = subset_samples(ps.fecal, sample_data(ps.fecal)$Time == "35") %>%
  prune_species(speciesSums(.) > 0, .)

out = ancombc(phyloseq = ps.fecal.35, formula = "Soil + Flu", 
              p_adj_method = "BH", zero_cut = 0.95, lib_cut = 1000, 
              group = "Soil", struc_zero = F, neg_lb = F, tol = 1e-5, 
              max_iter = 100, conserve = TRUE, alpha = 0.05, global = TRUE)

out.fecal = out
res = out.fecal$res
tab_q.fecal = res$q

col_name = c("Ctrl-Pine_fecal", "Ctrl-River_fecal", "Ctrl-Road_fecal", "HI-PR8_fecal")
colnames(tab_q.fecal) = col_name
tab_q.fecal
```

```{r}
tab = left_join(tab_q.lung, tab_q.soil, by = "ASV")

PineDF =  subset(tab, tab$`Ctrl-Pine_lung` < 0.05 & tab$`Ctrl-Pine_soil` < 0.05)
RiverDF =  subset(tab, tab$`Ctrl-River_lung` < 0.05 & tab$`Ctrl-River_soil` < 0.05)
RoadDF =  subset(tab, tab$`Ctrl-Road_lung` < 0.05 & tab$`Ctrl-Road_soil` < 0.05)
```

```{r, phylum barplot}
library(dplyr)

ps.phylum = tax_glom(ps.lung.r, "phylum")

psC1 = subset_samples(ps.phylum, sample_data(ps.phylum)$Soil == "Pine") 

genusDF = as.data.frame(cbind(t(otu_table(psC1, taxa_are_rows = FALSE)), tax_table(psC1)))

rownames(genusDF) = genusDF$phylum
genusDF$domain = NULL
genusDF$phylum=NULL
genusDF$class=NULL
genusDF$order=NULL
genusDF$family=NULL
genusDF$genus=NULL
genusDF$species = NULL
genusDF = mutate_all(genusDF, function(x) as.numeric(as.character(x)))

genusDF$C1 = rowMeans(genusDF)

PineDF = genusDF$C1

psC1 = subset_samples(ps.phylum, sample_data(ps.phylum)$Soil == "Ctrl") %>%
  tax_glom(., "phylum")

genusDF = as.data.frame(cbind(t(otu_table(psC1, taxa_are_rows = FALSE)), tax_table(psC1)))

rownames(genusDF) = genusDF$phylum
genusDF$domain = NULL
genusDF$phylum=NULL
genusDF$class=NULL
genusDF$order=NULL
genusDF$family=NULL
genusDF$genus=NULL
genusDF$species = NULL
genusDF = mutate_all(genusDF, function(x) as.numeric(as.character(x)))

genusDF$C1 = rowMeans(genusDF)

CtrlDF = genusDF$C1

psC1 = subset_samples(ps.phylum, sample_data(ps.phylum)$Soil == "River") %>%
  tax_glom(., "phylum")

genusDF = as.data.frame(cbind(t(otu_table(psC1, taxa_are_rows = FALSE)), tax_table(psC1)))

rownames(genusDF) = genusDF$phylum
genusDF$domain = NULL
genusDF$phylum=NULL
genusDF$class=NULL
genusDF$order=NULL
genusDF$family=NULL
genusDF$genus=NULL
genusDF$species = NULL
genusDF = mutate_all(genusDF, function(x) as.numeric(as.character(x)))

genusDF$C1 = rowMeans(genusDF)

RiverDF = genusDF$C1

psC1 = subset_samples(ps.phylum, sample_data(ps.phylum)$Soil == "Road") %>%
  tax_glom(., "phylum")

genusDF = as.data.frame(cbind(t(otu_table(psC1, taxa_are_rows = FALSE)), tax_table(psC1)))

rownames(genusDF) = genusDF$phylum
genusDF$domain = NULL
genusDF$phylum=NULL
genusDF$class=NULL
genusDF$order=NULL
genusDF$family=NULL
genusDF$genus=NULL
genusDF$species = NULL
genusDF = mutate_all(genusDF, function(x) as.numeric(as.character(x)))

genusDF$C1 = rowMeans(genusDF)

RoadDF = genusDF$C1

LungAvgDF = cbind(CtrlDF, PineDF, RiverDF, RoadDF) %>%
  data.frame(.)

rownames(LungAvgDF) = rownames(genusDF)

LungAvgDF$Avg = rowMeans(LungAvgDF)

figLung = top_n(LungAvgDF, 5, Avg) 

Other = 1 - colSums(figLung)

figLung = rbind(figLung, Other)

figLung$tax = rownames(figLung)

figLung[6,6] = "Other"

figLung = select(figLung, -Avg) %>%
  dplyr::rename('Ctrl' = CtrlDF,
                'Pine' = PineDF,
                'River' = RiverDF,
                'Road' = RoadDF)

#########Fecal Samples
library(dplyr)

ps.phylum = subset_samples(ps.fecal.r, sample_data(ps.fecal.r)$Time == "35") %>%
  tax_glom(., "phylum")

psC1 = subset_samples(ps.phylum, sample_data(ps.phylum)$Soil == "Pine") 

genusDF = as.data.frame(cbind(t(otu_table(psC1, taxa_are_rows = FALSE)), tax_table(psC1)))

rownames(genusDF) = genusDF$phylum
genusDF$domain = NULL
genusDF$phylum=NULL
genusDF$class=NULL
genusDF$order=NULL
genusDF$family=NULL
genusDF$genus=NULL
genusDF$species = NULL
genusDF = mutate_all(genusDF, function(x) as.numeric(as.character(x)))

genusDF$C1 = rowMeans(genusDF)

PineDF = genusDF$C1

psC1 = subset_samples(ps.phylum, sample_data(ps.phylum)$Soil == "Ctrl") %>%
  tax_glom(., "phylum")

genusDF = as.data.frame(cbind(t(otu_table(psC1, taxa_are_rows = FALSE)), tax_table(psC1)))

rownames(genusDF) = genusDF$phylum
genusDF$domain = NULL
genusDF$phylum=NULL
genusDF$class=NULL
genusDF$order=NULL
genusDF$family=NULL
genusDF$genus=NULL
genusDF$species = NULL
genusDF = mutate_all(genusDF, function(x) as.numeric(as.character(x)))

genusDF$C1 = rowMeans(genusDF)

CtrlDF = genusDF$C1

psC1 = subset_samples(ps.phylum, sample_data(ps.phylum)$Soil == "River") %>%
  tax_glom(., "phylum")

genusDF = as.data.frame(cbind(t(otu_table(psC1, taxa_are_rows = FALSE)), tax_table(psC1)))

rownames(genusDF) = genusDF$phylum
genusDF$domain = NULL
genusDF$phylum=NULL
genusDF$class=NULL
genusDF$order=NULL
genusDF$family=NULL
genusDF$genus=NULL
genusDF$species = NULL
genusDF = mutate_all(genusDF, function(x) as.numeric(as.character(x)))

genusDF$C1 = rowMeans(genusDF)

RiverDF = genusDF$C1

psC1 = subset_samples(ps.phylum, sample_data(ps.phylum)$Soil == "Road") %>%
  tax_glom(., "phylum")

genusDF = as.data.frame(cbind(t(otu_table(psC1, taxa_are_rows = FALSE)), tax_table(psC1)))

rownames(genusDF) = genusDF$phylum
genusDF$domain = NULL
genusDF$phylum=NULL
genusDF$class=NULL
genusDF$order=NULL
genusDF$family=NULL
genusDF$genus=NULL
genusDF$species = NULL
genusDF = mutate_all(genusDF, function(x) as.numeric(as.character(x)))

genusDF$C1 = rowMeans(genusDF)

RoadDF = genusDF$C1

FecalAvgDF = cbind(CtrlDF, PineDF, RiverDF, RoadDF) %>%
  data.frame(.)

rownames(FecalAvgDF) = rownames(genusDF)

FecalAvgDF$Avg = rowMeans(FecalAvgDF)

figFecal = top_n(FecalAvgDF, 5, Avg) 

Other = 1 - colSums(figFecal)

figFecal = rbind(figFecal, Other)

figFecal$tax = rownames(figFecal)

figFecal[6,6] = "Other"

figFecal = select(figFecal, -Avg) %>%
  dplyr::rename('Ctrl' = CtrlDF,
                'Pine' = PineDF,
                'River' = RiverDF,
                'Road' = RoadDF)

#########Soil Samples
library(dplyr)

ps.phylum = subset_samples(ps.soil.r, sample_data(ps.soil.r)$Time == "35") %>%
  tax_glom(., "phylum")

psC1 = subset_samples(ps.phylum, sample_data(ps.phylum)$Soil == "Pine") 

genusDF = as.data.frame(cbind(t(otu_table(psC1, taxa_are_rows = FALSE)), tax_table(psC1)))

rownames(genusDF) = genusDF$phylum
genusDF$domain = NULL
genusDF$phylum=NULL
genusDF$class=NULL
genusDF$order=NULL
genusDF$family=NULL
genusDF$genus=NULL
genusDF$species = NULL
genusDF = mutate_all(genusDF, function(x) as.numeric(as.character(x)))

genusDF$C1 = rowMeans(genusDF)

PineDF = genusDF$C1

psC1 = subset_samples(ps.phylum, sample_data(ps.phylum)$Soil == "Ctrl") %>%
  tax_glom(., "phylum")

genusDF = as.data.frame(cbind(t(otu_table(psC1, taxa_are_rows = FALSE)), tax_table(psC1)))

rownames(genusDF) = genusDF$phylum
genusDF$domain = NULL
genusDF$phylum=NULL
genusDF$class=NULL
genusDF$order=NULL
genusDF$family=NULL
genusDF$genus=NULL
genusDF$species = NULL
genusDF = mutate_all(genusDF, function(x) as.numeric(as.character(x)))

genusDF$C1 = rowMeans(genusDF)

CtrlDF = genusDF$C1

psC1 = subset_samples(ps.phylum, sample_data(ps.phylum)$Soil == "River") %>%
  tax_glom(., "phylum")

genusDF = as.data.frame(cbind(t(otu_table(psC1, taxa_are_rows = FALSE)), tax_table(psC1)))

rownames(genusDF) = genusDF$phylum
genusDF$domain = NULL
genusDF$phylum=NULL
genusDF$class=NULL
genusDF$order=NULL
genusDF$family=NULL
genusDF$genus=NULL
genusDF$species = NULL
genusDF = mutate_all(genusDF, function(x) as.numeric(as.character(x)))

genusDF$C1 = rowMeans(genusDF)

RiverDF = genusDF$C1

psC1 = subset_samples(ps.phylum, sample_data(ps.phylum)$Soil == "Road") %>%
  tax_glom(., "phylum")

genusDF = as.data.frame(cbind(t(otu_table(psC1, taxa_are_rows = FALSE)), tax_table(psC1)))

rownames(genusDF) = genusDF$phylum
genusDF$domain = NULL
genusDF$phylum=NULL
genusDF$class=NULL
genusDF$order=NULL
genusDF$family=NULL
genusDF$genus=NULL
genusDF$species = NULL
genusDF = mutate_all(genusDF, function(x) as.numeric(as.character(x)))

genusDF$C1 = rowMeans(genusDF)

RoadDF = genusDF$C1

SoilAvgDF = cbind(CtrlDF, PineDF, RiverDF, RoadDF) %>%
  data.frame(.)

rownames(SoilAvgDF) = rownames(genusDF)

SoilAvgDF$Avg = rowMeans(SoilAvgDF)

figSoil = top_n(SoilAvgDF, 5, Avg) 

Other = 1 - colSums(figSoil)

figSoil = rbind(figSoil, Other)

figSoil$tax = rownames(figSoil)

figSoil[6,6] = "Other"

figSoil = select(figSoil, -Avg) %>%
  dplyr::rename('Ctrl' = CtrlDF,
                'Pine' = PineDF,
                'River' = RiverDF,
                'Road' = RoadDF)

#############Put it together
library(reshape2)
fLung = melt(figLung)
fLung$Sample = "Lung"
fSoil = melt(figSoil)
fSoil$Sample = "Soil"
fFecal = melt(figFecal)
fFecal$Sample = "Fecal"

f = rbind(fLung, fSoil, fFecal)
unique(f$tax)
library(RColorBrewer)

mycolors = c("#666666","#4A72A6","#B75F49","#5ab4ac","#e6550d","#EC83BA","#48A462")#,"#e6550d","#B75F49","#5ab4ac","#666666")

f$tax = factor(f$tax, levels=c("Other","Proteobacteria", "Verrucomicrobiota", "Acidobacteriota", "Actinobacteriota", "Bacteroidota", "Firmicutes"))

p=ggplot(f, aes(fill = tax, y=value, x=variable)) +
  geom_bar(position="stack", stat="identity", color = "black", size = 0.25) +
  scale_fill_manual(values = mycolors) +
  xlab(NULL) +
  ylab("Mean relative abundance") +
  labs(fill = "Phylum") +
  theme_minimal() +
  scale_y_continuous(limits = c(0,1), expand = c(0, 0))+
theme(legend.text.align = 0) +
  facet_grid(~Sample)
```


```{r}
install.package("phylofactor")
library(phylofactor)

filter <- phyloseq::genefilter_sample(ps.lung, filterfun_sample(function(x) x >= 1), A = 0.05*nsamples(ps.lung))

ps.PF <- prune_taxa(filter, ps.lung)

sample = otu_table(ps.PF) %>%
  t(.) %>%
  as.data.frame(.) 

tree = phy_tree(ps.PF)

tree = unroot(tree)

tree <- makeNodeLabel(tree, method="number", prefix='n')

sampleDF = sample_data(ps.PF) %>%
  data.frame(.) %>%
  select(., Soil, Flu)

sampleDF$Soil = as.factor(sampleDF$Soil)
sampleDF$Flu = as.factor(sampleDF$Flu)

taxtab = as.data.frame(t(tax_table(ps.PF)))

PF=PhyloFactor(t(otu_table(ps.PF)),phy_tree(ps.PF),sample_data(ps.PF),nfactors=10, frmla = Data ~ Soil*Flu,  ncores = 12)

PF


```



