---
title: "Mouse Data Processing"
output: html_document
---

```{r}
MetaData = read.csv("~/mouseenv/Ottman/SraRunTable.txt", sep = ",")
```

```{bash, unzip files for DADA2}
gunzip /shared_space/Ottman_seq/*/*.gz
```

```{r, load dada2 and check version}
library(dada2); packageVersion("dada2")
```

```{r}
fpath = "/shared_space/Ottman_seq/R1"
rpath = "/shared_space/Ottman_seq/R2"

fnFs = sort(list.files(fpath, full.names = T))
fnRs = sort(list.files(rpath, full.names = T))

sample.names = sapply(strsplit(basename(fnFs), "_"), `[`, 1)
```

```{r}
plotQualityProfile(fnFs[4:5])
```

```{r}
plotQualityProfile(fnRs[1:2])
```

```{r}
filtFs <- file.path(fpath, "filtered", paste0(sample.names, "_F_filt.fastq.gz"))
filtRs <- file.path(rpath, "filtered", paste0(sample.names, "_R_filt.fastq.gz"))
names(filtFs) <- sample.names
names(filtRs) <- sample.names
```

```{bash, check that primers have been removed}
#F primer seq AGAGTTTGATCMTGGCTCAG
#R primer seq GTATTACCGCGGCTGCTG
head /shared_space/Ottman_seq/R1/SRR7205348_pass_1.fastq

##barcodes need to be removed using filterAndTrim
```


```{r}
out <- filterAndTrim(fnFs, filtFs, fnRs, filtRs, truncLen = c(290, 275), trimLeft = c(20,18),
              maxN=0, maxEE=c(5,5), truncQ=2, rm.phix=TRUE,
              compress=TRUE, multithread=TRUE)
head(out)
```

```{r}
errF = learnErrors(filtFs, multithread = T)
errR = learnErrors(filtRs, multithread = T)
plotErrors(errF, nominalQ = T)
```

```{r}
#get new filter names because some files dropped out
filF = sort(list.files("/shared_space/Ottman_seq/R1/filtered/", full.names = T))
filR = sort(list.files("/shared_space/Ottman_seq/R2/filtered/", full.names = T))

dadaFs = dada(filF, err = errF, multithread = T)
dadaRs = dada(filR, err = errR, multithread = T)
```

```{r}
mergers <- mergePairs(dadaFs, filF, dadaRs, filR, verbose=TRUE)
seqtab <- makeSequenceTable(mergers)
table(nchar(getSequences(seqtab)))

seqtab.trim <- seqtab[,nchar(colnames(seqtab)) %in% 480:502]
seqtab.nochim = removeBimeraDenovo(seqtab.trim, method = "consensus", multithread = T, verbose = T)

bim = isBimeraDenovoTable(seqtab.trim)
table(bim)
sum(seqtab.nochim)/sum(seqtab.trim)
dada2:::pfasta(head(getSequences(seqtab.trim)[bim]))
```

```{r}
library(DECIPHER)
dna = DNAStringSet(getSequences(seqtab.nochim))
load("/shared_space/SILVA_SSU_r138_2019.RData")
ids <- IdTaxa(dna, trainingSet, strand="top", processors=NULL, verbose=FALSE) # use all processors
ranks <- c("domain", "phylum", "class", "order", "family", "genus", "species") # ranks of interest
# Convert the output object of class "Taxa" to a matrix analogous to the output from assignTaxonomy
taxid <- t(sapply(ids, function(x) {
        m <- match(ranks, x$rank)
        taxa <- x$taxon[m]
        taxa[startsWith(taxa, "unclassified_")] <- NA
        taxa
}))
colnames(taxid) <- ranks; rownames(taxid) <- getSequences(seqtab.nochim)
head(taxid)
```

```{r}
library(phyloseq); packageVersion("phyloseq")
library(Biostrings); packageVersion("Biostrings")

MetaData1 = MetaData[-92, ]

samples.out <- rownames(seqtab.nochim)
rownames(MetaData1) <- samples.out

ps <- phyloseq(otu_table(seqtab.nochim, taxa_are_rows=FALSE), 
               sample_data(MetaData1), 
               tax_table(taxid))
```

```{r}
library(phangorn)

uniques=getUniques(seqtab.nochim)
sequences=names(uniques)
uniquesToFasta(uniques,fout = "~/mouseenv/Ottman/mouse.fasta", ids=sequences)
```

```{bash}
head mouse1.fasta
```


```{bash}
wget  "https://raw.github.com/smirarab/sepp-refs/master/gg/sepp-package.tar.bz"

tar xvfj sepp-package.tar.bz

cd sepp-package/sepp
python setup.py config -c

./sepp-package/run-sepp.sh ./Ottman/mouse1.fasta Ottman
```

```{r, view tree file with ggtree}
if (!requireNamespace("BiocManager", quietly = TRUE))
    install.packages("BiocManager")

BiocManager::install("ggtree")

library(ape)

Ottman.tree = read.tree(file = "~/mouseenv/mouse_placement.tog.relabelled.tre") %>%
  keep.tip(., sequences)
```

```{r, merge tree with phyloseq object}
ps.Ott = merge_phyloseq(ps, Ottman.tree)

dna <- Biostrings::DNAStringSet(taxa_names(ps.Ott))
names(dna) <- taxa_names(ps.Ott)
ps.Ott <- merge_phyloseq(ps.Ott, dna)
taxa_names(ps.Ott) <- paste0("ASV", seq(ntaxa(ps.Ott)))
```

